{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 524,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from collections import Counter\n",
    "from pathlib import Path\n",
    "data_location = Path(\"../Data/Tableau_Terra_workspace\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create the terra workspace \n",
    "Read all these weird files and try to merge them to the best of my capabilities:\n",
    "\n",
    "1. Read and merge CLF with Premium\n",
    "2. Read and merge Panel data\n",
    "3. Read RNA data\n",
    "4. Read and merge WES data - UPCOMMING\n",
    "5. Apply loop through rows\n",
    "6. Duplicate rows and add conditions\n",
    "7. Check number of datapoints\n",
    "\n",
    "Note, losing alot of information due to inconsistencies in the datasets. Another note this workflow eats up memory, because I keep meta data open, just something to note\n",
    "\n",
    "### 1 Read and merge CLF with Premium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 796,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CLF data and cleaning\n",
    "CLF_data_meta = pd.read_csv(data_location / \"CLF_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "CLF_data = CLF_data_meta[[\"Cell Line\", \"PT-ID\", \"SM-ID\", \"Status\", \"Resolution\", \"T/N\", \"Primary Disease\", \"Tissue site\", \"Tumor type\", \"CELL_LINE_ID(raw)\", \"Media type\", \"Growth Properties\", \"Flask coating\", \"Starting Material Type\", \"Fail Mode\"]]\n",
    "CLF_data = CLF_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "CLF_data = CLF_data.rename(columns={\"Cell Line\": \"Cell Line Starting Sample\", \"Tissue site\": \"Tissue Site\", \"Tumor type\": \"Tumor Type\", \"CELL_LINE_ID(raw)\":\"Cell Line\", \"Growth Properties\": \"Dimension\"})\n",
    "CLF_data[\"Dataset\"] = \"CLF\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 805,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3062: DtypeWarning: Columns (5,33,80,95) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "#Premium data and cleaning\n",
    "Premium_data_meta = pd.read_csv(data_location / \"Premium_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "Premium_data = Premium_data_meta[[\"Cell Line Starting Sample\", \"PT_ID\", \"Passage Sample\", \"JIRA Status\", \"Resolution\", \"T/N\", \"Primary disease\", \"Tissue Site\", \"Tumor Type\", \"CELL_LINE_ID\", \"Growth Medium\", \"Growth Pattern\", \"ECM Bio Coatings\", \"Root Material Type\"]]\n",
    "Premium_data = Premium_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "Premium_data = Premium_data.rename(columns={\"PT_ID\": \"PT-ID\", \"Passage Sample\": \"SM-ID\", \"JIRA Status\": \"Status\", \"Primary disease\": \"Primary Disease\", \"CELL_LINE_ID\":\"Cell Line\", \"Growth Medium\":\"Media type\", \"Growth Pattern\":\"Dimension\", \"ECM Bio Coatings\":\"Flask coating\", \"Root Material Type\":\"Starting Material Type\"})\n",
    "Premium_data[\"Fail Mode\"] = 'Not Reported'\n",
    "Premium_data[\"Dataset\"] = \"Premium\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1057,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assayed Sample</th>\n",
       "      <th>Assay Date</th>\n",
       "      <th>Assay Level</th>\n",
       "      <th>Assay Result</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>Basal Medium</th>\n",
       "      <th>Batch Created</th>\n",
       "      <th>Batch ID</th>\n",
       "      <th>Batch Last Updated</th>\n",
       "      <th>Batch Min Passage</th>\n",
       "      <th>...</th>\n",
       "      <th>% FBS</th>\n",
       "      <th>Primary Disease RAW</th>\n",
       "      <th>PT_ID</th>\n",
       "      <th>Receipt Date</th>\n",
       "      <th>Root Material Type</th>\n",
       "      <th>Root Sample Barcode</th>\n",
       "      <th>Sample Type</th>\n",
       "      <th>TC-treated Surface</th>\n",
       "      <th>Tissue Site</th>\n",
       "      <th>Tumor Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2/2018 10:35:14 AM</td>\n",
       "      <td>BP-94398</td>\n",
       "      <td>1/8/2019 9:53:20 AM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lung cancer</td>\n",
       "      <td>PT-1GMQ3</td>\n",
       "      <td>12/12/2014 11:40:33 AM</td>\n",
       "      <td>Cells:Cell Line, Viable</td>\n",
       "      <td>SM-6ER9E</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Pleural Cavity</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2/2018 10:35:14 AM</td>\n",
       "      <td>BP-94398</td>\n",
       "      <td>1/8/2019 9:53:20 AM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lung cancer</td>\n",
       "      <td>PT-1GMQ3</td>\n",
       "      <td>12/12/2014 11:40:33 AM</td>\n",
       "      <td>Cells:Cell Line, Viable</td>\n",
       "      <td>SM-6ER9E</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Pleural Cavity</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2/2018 10:35:14 AM</td>\n",
       "      <td>BP-94398</td>\n",
       "      <td>1/8/2019 9:53:20 AM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lung cancer</td>\n",
       "      <td>PT-1GMQ3</td>\n",
       "      <td>12/12/2014 11:40:33 AM</td>\n",
       "      <td>Cells:Cell Line, Viable</td>\n",
       "      <td>SM-6ER9E</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Pleural Cavity</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SM-HB43Z</td>\n",
       "      <td>5/16/2018 12:00:00 AM</td>\n",
       "      <td>1st Verification</td>\n",
       "      <td>Pass</td>\n",
       "      <td>In-line genomic QC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2/2018 10:35:14 AM</td>\n",
       "      <td>BP-94398</td>\n",
       "      <td>1/8/2019 9:53:20 AM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lung cancer</td>\n",
       "      <td>PT-1GMQ3</td>\n",
       "      <td>12/12/2014 11:40:33 AM</td>\n",
       "      <td>Cells:Cell Line, Viable</td>\n",
       "      <td>SM-6ER9E</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Pleural Cavity</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2/2018 10:35:14 AM</td>\n",
       "      <td>BP-94398</td>\n",
       "      <td>1/8/2019 9:53:20 AM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lung cancer</td>\n",
       "      <td>PT-1GMQ3</td>\n",
       "      <td>12/12/2014 11:40:33 AM</td>\n",
       "      <td>Cells:Cell Line, Viable</td>\n",
       "      <td>SM-6ER9E</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Pleural Cavity</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9296</th>\n",
       "      <td>SM-I6UNM</td>\n",
       "      <td>10/11/2018 12:00:00 AM</td>\n",
       "      <td>1st Verification</td>\n",
       "      <td>Pass</td>\n",
       "      <td>ELISA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/2/2018 12:29:21 PM</td>\n",
       "      <td>BP-102837</td>\n",
       "      <td>6/24/2019 10:44:30 AM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Renal Translocation Carcinoma</td>\n",
       "      <td>PT-2DT8F</td>\n",
       "      <td>7/30/2018 3:21:07 PM</td>\n",
       "      <td>Tissue:Fresh Tissue</td>\n",
       "      <td>SM-HP7W3</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Kidney</td>\n",
       "      <td>Primary</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9297</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...</td>\n",
       "      <td>9/27/2018 3:17:40 PM</td>\n",
       "      <td>BP-101506</td>\n",
       "      <td>2/1/2019 12:28:07 PM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mixed adeno-neuroendocrine (MANEC) of the GE j...</td>\n",
       "      <td>PT-2HF7A</td>\n",
       "      <td>9/27/2018 10:26:21 AM</td>\n",
       "      <td>Tissue:Fresh Tissue</td>\n",
       "      <td>SM-HPADL</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9298</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...</td>\n",
       "      <td>9/27/2018 3:17:40 PM</td>\n",
       "      <td>BP-101506</td>\n",
       "      <td>2/1/2019 12:28:07 PM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mixed adeno-neuroendocrine (MANEC) of the GE j...</td>\n",
       "      <td>PT-2HF7A</td>\n",
       "      <td>9/27/2018 10:26:21 AM</td>\n",
       "      <td>Tissue:Fresh Tissue</td>\n",
       "      <td>SM-HPADL</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9299</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...</td>\n",
       "      <td>9/27/2018 3:17:40 PM</td>\n",
       "      <td>BP-101506</td>\n",
       "      <td>2/1/2019 12:28:07 PM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mixed adeno-neuroendocrine (MANEC) of the GE j...</td>\n",
       "      <td>PT-2HF7A</td>\n",
       "      <td>9/27/2018 10:26:21 AM</td>\n",
       "      <td>Tissue:Fresh Tissue</td>\n",
       "      <td>SM-HPADL</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9300</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...</td>\n",
       "      <td>9/27/2018 3:17:40 PM</td>\n",
       "      <td>BP-101506</td>\n",
       "      <td>2/1/2019 12:28:07 PM</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mixed adeno-neuroendocrine (MANEC) of the GE j...</td>\n",
       "      <td>PT-2HF7A</td>\n",
       "      <td>9/27/2018 10:26:21 AM</td>\n",
       "      <td>Tissue:Fresh Tissue</td>\n",
       "      <td>SM-HPADL</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>TC-treated polystyrene</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9301 rows Ã— 114 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Assayed Sample              Assay Date       Assay Level Assay Result  \\\n",
       "0               NaN                     NaN               NaN          NaN   \n",
       "1               NaN                     NaN               NaN          NaN   \n",
       "2               NaN                     NaN               NaN          NaN   \n",
       "3          SM-HB43Z   5/16/2018 12:00:00 AM  1st Verification         Pass   \n",
       "4               NaN                     NaN               NaN          NaN   \n",
       "...             ...                     ...               ...          ...   \n",
       "9296       SM-I6UNM  10/11/2018 12:00:00 AM  1st Verification         Pass   \n",
       "9297            NaN                     NaN               NaN          NaN   \n",
       "9298            NaN                     NaN               NaN          NaN   \n",
       "9299            NaN                     NaN               NaN          NaN   \n",
       "9300            NaN                     NaN               NaN          NaN   \n",
       "\n",
       "              Assay Type                                       Basal Medium  \\\n",
       "0                    NaN                                                NaN   \n",
       "1                    NaN                                                NaN   \n",
       "2                    NaN                                                NaN   \n",
       "3     In-line genomic QC                                                NaN   \n",
       "4                    NaN                                                NaN   \n",
       "...                  ...                                                ...   \n",
       "9296               ELISA                                                NaN   \n",
       "9297                 NaN  OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...   \n",
       "9298                 NaN  OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...   \n",
       "9299                 NaN  OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...   \n",
       "9300                 NaN  OPAC:  OPAC 39.0 %,OPAC: RspondinI CM 10.0 %,O...   \n",
       "\n",
       "              Batch Created   Batch ID     Batch Last Updated  \\\n",
       "0      2/2/2018 10:35:14 AM   BP-94398    1/8/2019 9:53:20 AM   \n",
       "1      2/2/2018 10:35:14 AM   BP-94398    1/8/2019 9:53:20 AM   \n",
       "2      2/2/2018 10:35:14 AM   BP-94398    1/8/2019 9:53:20 AM   \n",
       "3      2/2/2018 10:35:14 AM   BP-94398    1/8/2019 9:53:20 AM   \n",
       "4      2/2/2018 10:35:14 AM   BP-94398    1/8/2019 9:53:20 AM   \n",
       "...                     ...        ...                    ...   \n",
       "9296  11/2/2018 12:29:21 PM  BP-102837  6/24/2019 10:44:30 AM   \n",
       "9297   9/27/2018 3:17:40 PM  BP-101506   2/1/2019 12:28:07 PM   \n",
       "9298   9/27/2018 3:17:40 PM  BP-101506   2/1/2019 12:28:07 PM   \n",
       "9299   9/27/2018 3:17:40 PM  BP-101506   2/1/2019 12:28:07 PM   \n",
       "9300   9/27/2018 3:17:40 PM  BP-101506   2/1/2019 12:28:07 PM   \n",
       "\n",
       "      Batch Min Passage  ... % FBS  \\\n",
       "0                   0.0  ...   NaN   \n",
       "1                   0.0  ...   NaN   \n",
       "2                   0.0  ...   NaN   \n",
       "3                   0.0  ...   NaN   \n",
       "4                   0.0  ...   NaN   \n",
       "...                 ...  ...   ...   \n",
       "9296                0.0  ...   NaN   \n",
       "9297                0.0  ...   NaN   \n",
       "9298                0.0  ...   NaN   \n",
       "9299                0.0  ...   NaN   \n",
       "9300                0.0  ...   NaN   \n",
       "\n",
       "                                    Primary Disease RAW     PT_ID  \\\n",
       "0                                           Lung cancer  PT-1GMQ3   \n",
       "1                                           Lung cancer  PT-1GMQ3   \n",
       "2                                           Lung cancer  PT-1GMQ3   \n",
       "3                                           Lung cancer  PT-1GMQ3   \n",
       "4                                           Lung cancer  PT-1GMQ3   \n",
       "...                                                 ...       ...   \n",
       "9296                      Renal Translocation Carcinoma  PT-2DT8F   \n",
       "9297  mixed adeno-neuroendocrine (MANEC) of the GE j...  PT-2HF7A   \n",
       "9298  mixed adeno-neuroendocrine (MANEC) of the GE j...  PT-2HF7A   \n",
       "9299  mixed adeno-neuroendocrine (MANEC) of the GE j...  PT-2HF7A   \n",
       "9300  mixed adeno-neuroendocrine (MANEC) of the GE j...  PT-2HF7A   \n",
       "\n",
       "                Receipt Date       Root Material Type  Root Sample Barcode  \\\n",
       "0     12/12/2014 11:40:33 AM  Cells:Cell Line, Viable             SM-6ER9E   \n",
       "1     12/12/2014 11:40:33 AM  Cells:Cell Line, Viable             SM-6ER9E   \n",
       "2     12/12/2014 11:40:33 AM  Cells:Cell Line, Viable             SM-6ER9E   \n",
       "3     12/12/2014 11:40:33 AM  Cells:Cell Line, Viable             SM-6ER9E   \n",
       "4     12/12/2014 11:40:33 AM  Cells:Cell Line, Viable             SM-6ER9E   \n",
       "...                      ...                      ...                  ...   \n",
       "9296    7/30/2018 3:21:07 PM      Tissue:Fresh Tissue             SM-HP7W3   \n",
       "9297   9/27/2018 10:26:21 AM      Tissue:Fresh Tissue             SM-HPADL   \n",
       "9298   9/27/2018 10:26:21 AM      Tissue:Fresh Tissue             SM-HPADL   \n",
       "9299   9/27/2018 10:26:21 AM      Tissue:Fresh Tissue             SM-HPADL   \n",
       "9300   9/27/2018 10:26:21 AM      Tissue:Fresh Tissue             SM-HPADL   \n",
       "\n",
       "      Sample Type      TC-treated Surface     Tissue Site  Tumor Type  \n",
       "0           Tumor  TC-treated polystyrene  Pleural Cavity  Metastatic  \n",
       "1           Tumor  TC-treated polystyrene  Pleural Cavity  Metastatic  \n",
       "2           Tumor  TC-treated polystyrene  Pleural Cavity  Metastatic  \n",
       "3           Tumor  TC-treated polystyrene  Pleural Cavity  Metastatic  \n",
       "4           Tumor  TC-treated polystyrene  Pleural Cavity  Metastatic  \n",
       "...           ...                     ...             ...         ...  \n",
       "9296        Tumor  TC-treated polystyrene          Kidney     Primary  \n",
       "9297        Tumor  TC-treated polystyrene           Liver  Metastatic  \n",
       "9298        Tumor  TC-treated polystyrene           Liver  Metastatic  \n",
       "9299        Tumor  TC-treated polystyrene           Liver  Metastatic  \n",
       "9300        Tumor  TC-treated polystyrene           Liver  Metastatic  \n",
       "\n",
       "[9301 rows x 114 columns]"
      ]
     },
     "execution_count": 1057,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Premium_data_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 909,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge CLF and Premium\n",
    "data = pd.concat([CLF_data, Premium_data])\n",
    "\n",
    "# Remove Normals\n",
    "data = data[data[\"T/N\"] == \"Tumor\"]\n",
    "\n",
    "# Cleanup some of columns\n",
    "data[\"SM-ID\"] = [re.split('\\r|  ',x)[0].strip(' \\t\\n\\r') if str(x) != 'nan' else x for x in list(data[\"SM-ID\"])]\n",
    "data[\"Resolution\"] = [str(x) for x in list(data[\"Resolution\"])]\n",
    "data[\"Cell Line\"] = [str(x) for x in list(data[\"Cell Line\"])]\n",
    "\n",
    "# Note Dropping lot of conditions\n",
    "Resolution_dict = {\n",
    "    'Terminated':'Terminated',\n",
    "    'In Process':'In Process',\n",
    "    'Verification':'In Process',\n",
    "    'Ambiguous':'Drop',\n",
    "    'Completed':'Completed',\n",
    "    'Verified Normal':'Verified Normal',\n",
    "    'Verified Tumor':'Verified Tumor',\n",
    "    'Clinical team follow up':'In Process',\n",
    "    'Other verified tumor models':'Verified Tumor',\n",
    "    'Primary sample verified normal':'Verified Normal',\n",
    "    'Blacklisted; FP Failure':'Drop',\n",
    "    'Low Purity Tumor':'Drop',\n",
    "    'Clinical team follow up,Primary sample verified normal':'Verified Normal',\n",
    "    'Clinical team follow up,Other verified tumor models':'Verified Tumor',\n",
    "    'nan':'nan',\n",
    "    'Abandoned':'Drop',\n",
    "}\n",
    "\n",
    "data[\"Resolution_modified\"] = data[\"Resolution\"]\n",
    "data = data.replace({\"Resolution_modified\": Resolution_dict})\n",
    "data = data[(data[\"Resolution_modified\"] != \"Drop\") & (data[\"Resolution_modified\"] != \"In Process\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2 Read and combined Panel data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 925,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-925-dde66bea5383>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n"
     ]
    }
   ],
   "source": [
    "# TWIST Panel data\n",
    "TWIST_samples_meta = pd.read_csv(data_location / \"PANCAN_TWIST_sample.tsv\", sep=\"\\t\")\n",
    "TWIST_pair_meta = pd.read_csv(data_location / \"PANCAN_TWIST_pair.tsv\", sep=\"\\t\")\n",
    "TWIST_samples = TWIST_samples_meta[[\"entity:sample_id\", \"external_id_validation\", \"participant\", \"patient_id\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\"]]\n",
    "TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n",
    "TWIST_pair = TWIST_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "\n",
    "TWIST_samples = TWIST_samples[TWIST_samples[\"entity:sample_id\"].isin(TWIST_pair[\"case_sample\"])]\n",
    "TWIST_samples = TWIST_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TWIST = pd.merge(TWIST_samples, TWIST_pair, on='case_sample')\n",
    "TWIST = TWIST.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TWIST[\"Dataset\"] = \"TWIST\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 926,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TSCA Panel data\n",
    "TSCA_samples_meta = pd.read_csv(data_location / \"TSCA_HCMI_sample.tsv\", sep=\"\\t\")\n",
    "TSCA_pair_meta = pd.read_csv(data_location / \"TSCA_HCMI_pair.tsv\", sep=\"\\t\")\n",
    "TSCA_samples = TSCA_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_validation\", \"external_id_validation\", \"participant\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\", \"tn_decision_clean\"]]\n",
    "TSCA_pair = TSCA_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "TSCA_pair = TSCA_pair.dropna()\n",
    "\n",
    "TSCA_samples = TSCA_samples[TSCA_samples[\"entity:sample_id\"].isin(TSCA_pair[\"case_sample\"])]\n",
    "TSCA_samples = TSCA_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TSCA = pd.merge(TSCA_samples, TSCA_pair, on='case_sample')\n",
    "TSCA = TSCA.drop(columns=[\"case_sample\"])\n",
    "TSCA = TSCA.rename(columns={\"bsp_sample_id_validation\":\"case_sample\"})\n",
    "TSCA = TSCA.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TSCA[\"Dataset\"] = \"TSCA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 936,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge TWIST and TSCA with Tableau_data\n",
    "Panel = pd.concat([TWIST, TSCA])\n",
    "\n",
    "#Adding PANEL column head\n",
    "Panel = Panel.rename(columns={\"patient_id\":\"PT-ID\"})\n",
    "Panel.columns = [\"PANEL_\" + x if i > 4 else x for i,x in enumerate(Panel.columns)]\n",
    "\n",
    "Panel[\"participant\"] = [str(x) for x in Panel[\"participant\"]]\n",
    "Panel[\"PANEL_tn_decision_clean\"] = [str(x) for x in Panel[\"PANEL_tn_decision_clean\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3 Read RNA data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 847,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA data\n",
    "RNA_meta = pd.read_csv(data_location / \"CCLF_RNA_2_0.tsv\", sep=\"\\t\")\n",
    "RNA = RNA_meta[[\"entity:sample_id\", \n",
    "                     \"bsp_sample_id_rna\",  \n",
    "                     \"participant\", \n",
    "                     \"sample_type\", \n",
    "                     \"fastq1\", \n",
    "                     \"fastq2\", \n",
    "                     \"bai_file_rna\", \n",
    "                     \"bam_file_rna\", \n",
    "                     \"fusion_predictions\", \n",
    "                     \"fusion_predictions_abridged\", \n",
    "                     \"rnaseqc_count_metrics\", \n",
    "                     \"rnaseqc_count_outputs\", \n",
    "                     \"rnaseqc_exon_counts\", \n",
    "                     \"rnaseqc_gene_counts\", \n",
    "                     \"rnaseqc_gene_rpkm\", \n",
    "                     \"rsem_genes\", \n",
    "                     \"rsem_isoforms\", \n",
    "                     \"star_bam_file\", \n",
    "                     \"star_bam_index\", \n",
    "                     \"star_chimeric_bam_index\", \n",
    "                     \"star_chimeric_junctions\", \n",
    "                     \"star_junctions\", \n",
    "                     \"star_junctions_pass1\", \n",
    "                     \"star_md_bam_file\", \n",
    "                     \"star_md_bam_index\", \n",
    "                     \"star_read_counts\",\n",
    "                    \"star_transcriptome_bam\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 848,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Adding RNA column head\n",
    "RNA.columns = [\"RNA_\" + x if i > 3 else x for i,x in enumerate(RNA.columns)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 873,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removing without SM-ID\n",
    "total_rows = len(RNA)\n",
    "RNA[\"bsp_sample_id_rna\"] = [str(x) for x in RNA[\"bsp_sample_id_rna\"]]\n",
    "RNA[\"participant\"] = [str(x) for x in RNA[\"participant\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4 Read WES data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WES data\n",
    "WES_samples_meta = pd.read_csv(data_location / \"CCLF_WES_sample.tsv\", sep=\"\\t\")\n",
    "WES_pair_meta = pd.read_csv(data_location / \"CCLF_WES_pair.tsv\", sep=\"\\t\")\n",
    "WES_samples = WES_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_capture\", \"external_id_capture\", \"participant\", \"sample_type\", \"formatted_bai_file\", \"formatted_bam_file\", \"cnv_calls\"]]\n",
    "\n",
    "# Pair is wierd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5 Loop through data samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1059,
   "metadata": {},
   "outputs": [],
   "source": [
    "def growth_check(x):\n",
    "    # Checking resolution and mapping back\n",
    "    x[\"Fail Mode\"] = [str(x) for x in x[\"Fail Mode\"]]\n",
    "    \n",
    "    mapping_dict = {\n",
    "        'Terminated No Growth' : 0,\n",
    "        'Terminated Unlisted': 0,\n",
    "        'Terminated Fungus Contamination': 0,\n",
    "        'Terminated Stopped Growing': 0,\n",
    "        'Terminated Fungus Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,Sample Quality': 0,\n",
    "        'Terminated Fibroblast Overgrowth': 0,\n",
    "        'Terminated Feeder Contamination,No Growth': 0,\n",
    "        'Verified Normal Fibroblast Overgrowth': 0,\n",
    "        'Verified Tumor Stopped Growing': 0,\n",
    "        'Terminated Feeder Contamination': 0,\n",
    "        'Verified Tumor Fingerprinting Failure': 0,\n",
    "        'Verified Tumor Unlisted': 0,\n",
    "        'Verified Normal Mouse Contamination': 0,\n",
    "        'Verified Normal Stopped Growing': 0,\n",
    "        'Verified Normal Unlisted': 0,\n",
    "        'Verified Normal Sample Purity': 0,\n",
    "        'Terminated Fingerprinting Failure': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,No Growth': 0,\n",
    "        'Terminated No Growth,Sample Quality': 0,\n",
    "        'Terminated Sample Quality': 0,\n",
    "        'Terminated Lab Error,Sample Purity': 0,\n",
    "        'Terminated Mouse Contamination': 0,\n",
    "        'Terminated Fibroblast Overgrowth,Mouse Contamination': 0,\n",
    "        'Verified Tumor No Growth': 0,\n",
    "        'Terminated No Growth After Thaw': 0,\n",
    "        'Terminated No Growth,Sample Purity,Sample Quality': 0,\n",
    "        'Terminated Lab Error': 0,\n",
    "        'Terminated Fingerprinting Failure,Lab Error': 0,\n",
    "        'Verified Normal No Growth': 0,\n",
    "        'Terminated No Growth,Sample Purity': 0,\n",
    "        'Terminated Sample Purity': 0,\n",
    "        'Verified Tumor Not Reported': 1,\n",
    "        'Completed Not Reported': 1,\n",
    "        'nan Not Reported': 'nan',\n",
    "        'Verified Normal Not Reported': 0,\n",
    "        'nan': 'nan',\n",
    "    }\n",
    "    \n",
    "    series = x[\"Resolution_modified\"] + \" \" + x[\"Fail Mode\"]\n",
    "    series = [str(x) for x in series]\n",
    "    return series"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1061,
   "metadata": {},
   "outputs": [],
   "source": [
    "series = growth_check(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1062,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'Terminated No Growth': 1441,\n",
       "         'Completed nan': 106,\n",
       "         'Terminated Unlisted': 94,\n",
       "         'Terminated Fungus Contamination': 74,\n",
       "         'Terminated Stopped Growing': 713,\n",
       "         'Terminated Fungus Contamination,No Growth': 2,\n",
       "         'Terminated Bacterial Contamination': 67,\n",
       "         'Terminated Bacterial Contamination,No Growth': 20,\n",
       "         'Terminated Bacterial Contamination,Fungus Contamination,Sample Quality': 3,\n",
       "         'Terminated Fibroblast Overgrowth': 20,\n",
       "         'Terminated Feeder Contamination,No Growth': 1,\n",
       "         'Verified Normal nan': 451,\n",
       "         'Verified Normal Fibroblast Overgrowth': 2,\n",
       "         'Verified Tumor nan': 419,\n",
       "         'Verified Tumor Stopped Growing': 12,\n",
       "         'Terminated Feeder Contamination': 1,\n",
       "         'Verified Tumor Fingerprinting Failure': 1,\n",
       "         'Verified Tumor Unlisted': 45,\n",
       "         'Verified Normal Mouse Contamination': 3,\n",
       "         'Verified Normal Stopped Growing': 9,\n",
       "         'Verified Normal Unlisted': 29,\n",
       "         'Verified Normal Sample Purity': 10,\n",
       "         'Terminated Fingerprinting Failure': 1,\n",
       "         'Terminated Bacterial Contamination,Fungus Contamination': 2,\n",
       "         'Terminated Bacterial Contamination,Fungus Contamination,No Growth': 14,\n",
       "         'Terminated No Growth,Sample Quality': 6,\n",
       "         'Terminated Sample Quality': 3,\n",
       "         'Terminated Lab Error,Sample Purity': 2,\n",
       "         'Terminated Mouse Contamination': 1,\n",
       "         'Terminated Fibroblast Overgrowth,Mouse Contamination': 2,\n",
       "         'Verified Tumor No Growth': 4,\n",
       "         'Terminated No Growth After Thaw': 1,\n",
       "         'Terminated No Growth,Sample Purity,Sample Quality': 1,\n",
       "         'Terminated Lab Error': 2,\n",
       "         'Terminated Fingerprinting Failure,Lab Error': 1,\n",
       "         'Verified Normal No Growth': 7,\n",
       "         'Terminated No Growth,Sample Purity': 1,\n",
       "         'Terminated Sample Purity': 3,\n",
       "         'Verified Tumor Not Reported': 1671,\n",
       "         'Terminated Not Reported': 3882,\n",
       "         'Completed Not Reported': 159,\n",
       "         'nan Not Reported': 1362,\n",
       "         'Verified Normal Not Reported': 2076})"
      ]
     },
     "execution_count": 1062,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(series)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1013,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n"
     ]
    }
   ],
   "source": [
    "spreadsheet = []\n",
    "empty_numpy_panel = np.empty([1,len(list(Panel)[5:])])\n",
    "empty_numpy_panel.fill(np.nan)\n",
    "empty_frame_panel = pd.DataFrame(empty_numpy_panel, columns=list(Panel)[5:])\n",
    "\n",
    "for i, cell_line in enumerate(data[\"Cell Line\"].unique()):\n",
    "    data_tmp = data[data[\"Cell Line\"] == cell_line]\n",
    "    # At least one that isn't Verified Normal\n",
    "    if all(data_tmp[\"Resolution_modified\"] == \"Verified Normal\") != True:\n",
    "        # First get the right data from Panel/RNA\n",
    "        if cell_line.endswith(\"T\"):\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line[:-1]]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line[:-1]]\n",
    "        else:\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line]\n",
    "\n",
    "        #Add Panel data by looping through\n",
    "        if not Panel_tmp.empty:\n",
    "            # Check if all are not VN\n",
    "            if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                # Check if multiple sequencing results for cell line\n",
    "                if len(Panel_tmp) == 1:\n",
    "                    print('yes')\n",
    "                    \n",
    "                    break\n",
    "                else:\n",
    "                    break\n",
    "            #else:\n",
    "                #Add empty columns\n",
    "        #else:\n",
    "            #Add empty columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1014,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_sample</th>\n",
       "      <th>external_id_validation</th>\n",
       "      <th>participant</th>\n",
       "      <th>PT-ID</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>PANEL_renamed_bai_file</th>\n",
       "      <th>PANEL_renamed_bam_file</th>\n",
       "      <th>PANEL_depth_of_cov_qc_result</th>\n",
       "      <th>PANEL_cnv_calls</th>\n",
       "      <th>PANEL_media</th>\n",
       "      <th>PANEL_tn_decision_clean</th>\n",
       "      <th>PANEL_filtered_variants</th>\n",
       "      <th>PANEL_mutect1_callstats</th>\n",
       "      <th>PANEL_mutect1_coveragewig</th>\n",
       "      <th>PANEL_mutect1_powerwig</th>\n",
       "      <th>PANEL_mutect1_vcf</th>\n",
       "      <th>PANEL_mutect2_vcf</th>\n",
       "      <th>PANEL_oncotated_maf_mutect1</th>\n",
       "      <th>PANEL_oncotated_maf_mutect2</th>\n",
       "      <th>PANEL_Dataset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1828</th>\n",
       "      <td>SM-D59IN</td>\n",
       "      <td>HG020T_primary</td>\n",
       "      <td>HG020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/e...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/e...</td>\n",
       "      <td>pass</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>primary</td>\n",
       "      <td>VT</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/8...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...</td>\n",
       "      <td>TSCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     case_sample external_id_validation participant PT-ID sample_type  \\\n",
       "1828    SM-D59IN         HG020T_primary       HG020   NaN       Tumor   \n",
       "\n",
       "                                 PANEL_renamed_bai_file  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/e...   \n",
       "\n",
       "                                 PANEL_renamed_bam_file  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/e...   \n",
       "\n",
       "     PANEL_depth_of_cov_qc_result  \\\n",
       "1828                         pass   \n",
       "\n",
       "                                        PANEL_cnv_calls PANEL_media  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...     primary   \n",
       "\n",
       "     PANEL_tn_decision_clean  \\\n",
       "1828                      VT   \n",
       "\n",
       "                                PANEL_filtered_variants  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/8...   \n",
       "\n",
       "                                PANEL_mutect1_callstats  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...   \n",
       "\n",
       "                              PANEL_mutect1_coveragewig  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...   \n",
       "\n",
       "                                 PANEL_mutect1_powerwig  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...   \n",
       "\n",
       "                                      PANEL_mutect1_vcf  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...   \n",
       "\n",
       "                                      PANEL_mutect2_vcf  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...   \n",
       "\n",
       "                            PANEL_oncotated_maf_mutect1  \\\n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...   \n",
       "\n",
       "                            PANEL_oncotated_maf_mutect2 PANEL_Dataset  \n",
       "1828  gs://fc-c23078b3-05b3-4158-ba8f-2b1eeb1bfa16/9...          TSCA  "
      ]
     },
     "execution_count": 1014,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Panel_tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1032,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0               Terminated No Growth\n",
       "330                              NaN\n",
       "331                              NaN\n",
       "332                              NaN\n",
       "333                              NaN\n",
       "                    ...             \n",
       "9274    Verified Normal Not Reported\n",
       "9275         Terminated Not Reported\n",
       "9278                nan Not Reported\n",
       "9295                nan Not Reported\n",
       "9296                nan Not Reported\n",
       "Length: 12724, dtype: object"
      ]
     },
     "execution_count": 1032,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[\"Resolution_modified\"] + \" \" + data[\"Fail Mode\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 948,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Not Reported', 'VT', 'nan', 'VN'], dtype=object)"
      ]
     },
     "execution_count": 948,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Panel[\"PANEL_tn_decision_clean\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 910,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'Terminated': 6358,\n",
       "         'Completed': 265,\n",
       "         'Verified Normal': 2587,\n",
       "         'Verified Tumor': 2152,\n",
       "         'nan': 1362})"
      ]
     },
     "execution_count": 910,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(data[\"Resolution_modified\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check if primary\n",
    "Lookup_table = {\"PRIMARY\":13,\n",
    "               \"P0\":12,\n",
    "               \"P1\":11,\n",
    "               \"P2\":10,\n",
    "               \"P3\":9,\n",
    "               \"P4\":8,\n",
    "               \"P5\":7,\n",
    "               \"P6\":6,\n",
    "               \"P7\":5,\n",
    "               \"P8\":4,\n",
    "               \"P9\":3,\n",
    "               \"P10\":2,\n",
    "               \"P11\":1,\n",
    "               \"P12\":0,}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.1 64-bit ('Broad_env': virtualenv)",
   "language": "python",
   "name": "python38164bitbroadenvvirtualenvcc06190595da421ea68a050a83be054b"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
