{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1274,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from pathlib import Path\n",
    "data_location = Path(\"../Data/Tableau_Terra_workspace\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create the terra workspace \n",
    "Read all these weird files and try to merge them to the best of my capabilities:\n",
    "\n",
    "1. Read and merge CLF with Premium\n",
    "2. Read and merge Panel data\n",
    "3. Read RNA data\n",
    "4. Read and merge WES data - UPCOMMING\n",
    "5. Apply loop through rows\n",
    "6. Duplicate rows and add conditions\n",
    "7. Check number of datapoints\n",
    "\n",
    "Note, losing alot of information due to inconsistencies in the datasets. Another note this workflow eats up memory, because I keep meta data open, just something to note\n",
    "\n",
    "### 1 Read and merge CLF with Premium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1152,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CLF data and cleaning\n",
    "CLF_data_meta = pd.read_csv(data_location / \"CLF_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "CLF_data = CLF_data_meta[[\"Cell Line\", \"PT-ID\", \"SM-ID\", \"Status\", \"Resolution\", \"T/N\", \"Primary Disease\", \"Tissue site\", \"Tumor type\", \"CELL_LINE_ID(raw)\", \"Media type\", \"Growth Properties\", \"Flask coating\", \"Starting Material Type\", \"Fail Mode\"]]\n",
    "CLF_data = CLF_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "CLF_data = CLF_data.rename(columns={\"Cell Line\": \"Cell Line Starting Sample\", \"Tissue site\": \"Tissue Site\", \"Tumor type\": \"Tumor Type\", \"CELL_LINE_ID(raw)\":\"Cell Line\", \"Growth Properties\": \"Dimension\"})\n",
    "CLF_data[\"Dataset\"] = \"CLF\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1153,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3062: DtypeWarning: Columns (5,33,80,95) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "#Premium data and cleaning\n",
    "Premium_data_meta = pd.read_csv(data_location / \"Premium_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "Premium_data = Premium_data_meta[[\"Cell Line Starting Sample\", \"PT_ID\", \"Passage Sample\", \"JIRA Status\", \"Resolution\", \"T/N\", \"Primary disease\", \"Tissue Site\", \"Tumor Type\", \"CELL_LINE_ID\", \"Growth Medium\", \"Growth Pattern\", \"ECM Bio Coatings\", \"Root Material Type\", \"Passage Sample Status\"]]\n",
    "Premium_data = Premium_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "Premium_data = Premium_data.rename(columns={\"PT_ID\": \"PT-ID\", \"Passage Sample\": \"SM-ID\", \"JIRA Status\": \"Status\", \"Primary disease\": \"Primary Disease\", \"CELL_LINE_ID\":\"Cell Line\", \"Growth Medium\":\"Media type\", \"Growth Pattern\":\"Dimension\", \"ECM Bio Coatings\":\"Flask coating\", \"Root Material Type\":\"Starting Material Type\", \"Passage Sample Status\":\"Fail Mode\"})\n",
    "Premium_data[\"Dataset\"] = \"Premium\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1154,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge CLF and Premium\n",
    "data = pd.concat([CLF_data, Premium_data])\n",
    "\n",
    "# Remove Normals\n",
    "data = data[data[\"T/N\"] == \"Tumor\"]\n",
    "\n",
    "# Cleanup some of columns\n",
    "data[\"SM-ID\"] = [re.split('\\r|  ',x)[0].strip(' \\t\\n\\r') if str(x) != 'nan' else x for x in list(data[\"SM-ID\"])]\n",
    "data[\"Resolution\"] = [str(x) for x in list(data[\"Resolution\"])]\n",
    "data[\"Cell Line\"] = [str(x) for x in list(data[\"Cell Line\"])]\n",
    "data[\"Media type\"] = [str(x) for x in list(data[\"Media type\"])]\n",
    "data = data[data[\"Media type\"] != \"nan\"]\n",
    "\n",
    "# Note Dropping lot of conditions\n",
    "Resolution_dict = {\n",
    "    'Terminated':'Terminated',\n",
    "    'In Process':'In Process',\n",
    "    'Verification':'In Process',\n",
    "    'Ambiguous':'Drop',\n",
    "    'Completed':'Completed',\n",
    "    'Verified Normal':'Verified Normal',\n",
    "    'Verified Tumor':'Verified Tumor',\n",
    "    'Clinical team follow up':'In Process',\n",
    "    'Other verified tumor models':'Verified Tumor',\n",
    "    'Primary sample verified normal':'Verified Normal',\n",
    "    'Blacklisted; FP Failure':'Drop',\n",
    "    'Low Purity Tumor':'Drop',\n",
    "    'Clinical team follow up,Primary sample verified normal':'Verified Normal',\n",
    "    'Clinical team follow up,Other verified tumor models':'Verified Tumor',\n",
    "    'nan':'nan',\n",
    "    'Abandoned':'Drop',\n",
    "}\n",
    "\n",
    "data[\"Resolution_modified\"] = data[\"Resolution\"]\n",
    "data = data.replace({\"Resolution_modified\": Resolution_dict})\n",
    "data = data[(data[\"Resolution_modified\"] != \"Drop\") & (data[\"Resolution_modified\"] != \"In Process\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2 Read and combined Panel data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 925,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-925-dde66bea5383>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n"
     ]
    }
   ],
   "source": [
    "# TWIST Panel data\n",
    "TWIST_samples_meta = pd.read_csv(data_location / \"PANCAN_TWIST_sample.tsv\", sep=\"\\t\")\n",
    "TWIST_pair_meta = pd.read_csv(data_location / \"PANCAN_TWIST_pair.tsv\", sep=\"\\t\")\n",
    "TWIST_samples = TWIST_samples_meta[[\"entity:sample_id\", \"external_id_validation\", \"participant\", \"patient_id\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\"]]\n",
    "TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n",
    "TWIST_pair = TWIST_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "\n",
    "TWIST_samples = TWIST_samples[TWIST_samples[\"entity:sample_id\"].isin(TWIST_pair[\"case_sample\"])]\n",
    "TWIST_samples = TWIST_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TWIST = pd.merge(TWIST_samples, TWIST_pair, on='case_sample')\n",
    "TWIST = TWIST.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TWIST[\"Dataset\"] = \"TWIST\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 926,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TSCA Panel data\n",
    "TSCA_samples_meta = pd.read_csv(data_location / \"TSCA_HCMI_sample.tsv\", sep=\"\\t\")\n",
    "TSCA_pair_meta = pd.read_csv(data_location / \"TSCA_HCMI_pair.tsv\", sep=\"\\t\")\n",
    "TSCA_samples = TSCA_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_validation\", \"external_id_validation\", \"participant\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\", \"tn_decision_clean\"]]\n",
    "TSCA_pair = TSCA_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "TSCA_pair = TSCA_pair.dropna()\n",
    "\n",
    "TSCA_samples = TSCA_samples[TSCA_samples[\"entity:sample_id\"].isin(TSCA_pair[\"case_sample\"])]\n",
    "TSCA_samples = TSCA_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TSCA = pd.merge(TSCA_samples, TSCA_pair, on='case_sample')\n",
    "TSCA = TSCA.drop(columns=[\"case_sample\"])\n",
    "TSCA = TSCA.rename(columns={\"bsp_sample_id_validation\":\"case_sample\"})\n",
    "TSCA = TSCA.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TSCA[\"Dataset\"] = \"TSCA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 936,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge TWIST and TSCA with Tableau_data\n",
    "Panel = pd.concat([TWIST, TSCA])\n",
    "\n",
    "#Adding PANEL column head\n",
    "Panel = Panel.rename(columns={\"patient_id\":\"PT-ID\"})\n",
    "Panel.columns = [\"PANEL_\" + x if i > 4 else x for i,x in enumerate(Panel.columns)]\n",
    "\n",
    "Panel[\"participant\"] = [str(x) for x in Panel[\"participant\"]]\n",
    "Panel[\"PANEL_tn_decision_clean\"] = [str(x) for x in Panel[\"PANEL_tn_decision_clean\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3 Read RNA data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 847,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA data\n",
    "RNA_meta = pd.read_csv(data_location / \"CCLF_RNA_2_0.tsv\", sep=\"\\t\")\n",
    "RNA = RNA_meta[[\"entity:sample_id\", \n",
    "                     \"bsp_sample_id_rna\",  \n",
    "                     \"participant\", \n",
    "                     \"sample_type\", \n",
    "                     \"fastq1\", \n",
    "                     \"fastq2\", \n",
    "                     \"bai_file_rna\", \n",
    "                     \"bam_file_rna\", \n",
    "                     \"fusion_predictions\", \n",
    "                     \"fusion_predictions_abridged\", \n",
    "                     \"rnaseqc_count_metrics\", \n",
    "                     \"rnaseqc_count_outputs\", \n",
    "                     \"rnaseqc_exon_counts\", \n",
    "                     \"rnaseqc_gene_counts\", \n",
    "                     \"rnaseqc_gene_rpkm\", \n",
    "                     \"rsem_genes\", \n",
    "                     \"rsem_isoforms\", \n",
    "                     \"star_bam_file\", \n",
    "                     \"star_bam_index\", \n",
    "                     \"star_chimeric_bam_index\", \n",
    "                     \"star_chimeric_junctions\", \n",
    "                     \"star_junctions\", \n",
    "                     \"star_junctions_pass1\", \n",
    "                     \"star_md_bam_file\", \n",
    "                     \"star_md_bam_index\", \n",
    "                     \"star_read_counts\",\n",
    "                    \"star_transcriptome_bam\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 848,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Adding RNA column head\n",
    "RNA.columns = [\"RNA_\" + x if i > 3 else x for i,x in enumerate(RNA.columns)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 873,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removing without SM-ID\n",
    "total_rows = len(RNA)\n",
    "RNA[\"bsp_sample_id_rna\"] = [str(x) for x in RNA[\"bsp_sample_id_rna\"]]\n",
    "RNA[\"participant\"] = [str(x) for x in RNA[\"participant\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4 Read WES data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WES data\n",
    "WES_samples_meta = pd.read_csv(data_location / \"CCLF_WES_sample.tsv\", sep=\"\\t\")\n",
    "WES_pair_meta = pd.read_csv(data_location / \"CCLF_WES_pair.tsv\", sep=\"\\t\")\n",
    "WES_samples = WES_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_capture\", \"external_id_capture\", \"participant\", \"sample_type\", \"formatted_bai_file\", \"formatted_bam_file\", \"cnv_calls\"]]\n",
    "\n",
    "# Pair is wierd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5 Loop through data samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1155,
   "metadata": {},
   "outputs": [],
   "source": [
    "def growth_check(x):\n",
    "    mapping_dict = {\n",
    "        'Terminated No Growth': 0,\n",
    "        'Terminated Unlisted': 0,\n",
    "        'Terminated Fungus Contamination': 0,\n",
    "        'Terminated Stopped Growing': 0,\n",
    "        'Terminated Fungus Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,Sample Quality': 0,\n",
    "        'Terminated Fibroblast Overgrowth': 0,\n",
    "        'Terminated Feeder Contamination,No Growth': 0,\n",
    "        'Verified Normal nan': 0,\n",
    "        'Verified Normal Fibroblast Overgrowth': 0,\n",
    "        'Verified Tumor nan': 1,\n",
    "        'Verified Tumor Stopped Growing': 0,\n",
    "        'Terminated Feeder Contamination': 0,\n",
    "        'Verified Tumor Fingerprinting Failure': 2,\n",
    "        'Verified Tumor Unlisted': 0,\n",
    "        'Verified Normal Mouse Contamination': 0,\n",
    "        'Verified Normal Stopped Growing': 0,\n",
    "        'Verified Normal Unlisted': 0,\n",
    "        'Verified Normal Sample Purity': 0,\n",
    "        'Terminated Fingerprinting Failure': 2,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,No Growth': 0,\n",
    "        'Terminated No Growth,Sample Quality': 0,\n",
    "        'Terminated Sample Quality': 0,\n",
    "        'Terminated Lab Error,Sample Purity': 0,\n",
    "        'Terminated Mouse Contamination': 0,\n",
    "        'Terminated Fibroblast Overgrowth,Mouse Contamination': 0,\n",
    "        'Verified Tumor No Growth': 0,\n",
    "        'Terminated No Growth After Thaw': 0,\n",
    "        'Terminated No Growth,Sample Purity,Sample Quality': 0,\n",
    "        'Terminated Lab Error': 0,\n",
    "        'Terminated Fingerprinting Failure,Lab Error': 0,\n",
    "        'Verified Normal No Growth': 0,\n",
    "        'Terminated No Growth,Sample Purity': 0,\n",
    "        'Terminated Sample Purity': 0,\n",
    "        'Verified Tumor Terminated': 0,\n",
    "        'Terminated Terminated': 0,\n",
    "        'Terminated Received/In Container': 0,\n",
    "        'Completed Terminated': 0,\n",
    "        'Completed Received/Out of Storage': 0,\n",
    "        'Terminated nan': 0,\n",
    "        'nan nan': 2,\n",
    "        'Verified Tumor Received/Out of Storage': 2,\n",
    "        'Verified Tumor Received/In Container': 2,\n",
    "        'Verified Normal Terminated': 0,\n",
    "        'Verified Normal Received/In Container': 2,\n",
    "        'Verified Normal Received/Out of Storage': 2,\n",
    "        'nan Terminated': 0,\n",
    "        'nan Received/In Container': 2,\n",
    "        'Terminated Received/Out of Storage': 0,\n",
    "        'nan Received/Out of Storage': 2,\n",
    "        'Completed nan': 1,\n",
    "    }\n",
    "    \n",
    "    # Checking resolution and mapping back\n",
    "    x[\"Fail Mode\"] = [str(x) for x in x[\"Fail Mode\"]]\n",
    "    \n",
    "    series = x[\"Resolution_modified\"] + \" \" + x[\"Fail Mode\"]\n",
    "    series = [str(x) for x in series]\n",
    "\n",
    "    x[\"Growth\"] = series\n",
    "    x = x.replace({\"Growth\": mapping_dict})\n",
    "\n",
    "    x = x[x[\"Growth\"] != 2]\n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1156,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = growth_check(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1292,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_score(x):\n",
    "    Lookup_table = {\n",
    "        \"PRIMARY\":19,\n",
    "        \"P0\":18,\n",
    "        \"P1\":17,\n",
    "        \"P2\":16,\n",
    "        \"P3\":15,\n",
    "        \"P4\":14,\n",
    "        \"P5\":13,\n",
    "        \"P6\":12,\n",
    "        \"P7\":11,\n",
    "        \"P8\":10,\n",
    "        \"P9\":9,\n",
    "        \"P10\":8,\n",
    "        \"P11\":7,\n",
    "        \"P12\":6,\n",
    "        \"P13\":5,\n",
    "        \"P14\":4,\n",
    "        \"P15\":3,\n",
    "        \"P16\":2,\n",
    "        \"P17\":1,\n",
    "    }\n",
    "\n",
    "    test = [x.upper().split(\"_\") for x in x[\"external_id_validation\"]]\n",
    "    \n",
    "    value = []\n",
    "    for t in test:\n",
    "        for key in t:\n",
    "            if key in Lookup_table:\n",
    "                value_tmp = Lookup_table[key]\n",
    "                break\n",
    "            else:\n",
    "                value_tmp = 0\n",
    "\n",
    "        value.append(value_tmp)\n",
    "        \n",
    "    x[\"test\"] = value\n",
    "    \n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1423,
   "metadata": {},
   "outputs": [],
   "source": [
    "empty_numpy_panel = np.empty([1,len(list(Panel)[5:])])\n",
    "empty_numpy_panel.fill(np.nan)\n",
    "empty_frame_panel = pd.DataFrame(empty_numpy_panel, columns=list(Panel)[5:])\n",
    "\n",
    "empty_numpy_RNA = np.empty([1,len(list(RNA)[4:])])\n",
    "empty_numpy_RNA.fill(np.nan)\n",
    "empty_frame_RNA = pd.DataFrame(empty_numpy_RNA, columns=list(RNA)[4:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1435,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_nan_panel(data_tmp, empty_numpy_panel):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_panel]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)\n",
    "\n",
    "def add_nan_RNA(data_tmp, empty_frame_RNA):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_RNA]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1508,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-1292-c83dd44c1f94>:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  x[\"test\"] = value\n"
     ]
    }
   ],
   "source": [
    "spreadsheet = []\n",
    "\n",
    "for i, cell_line in enumerate(data[\"Cell Line\"].unique()):\n",
    "    data_tmp = data[data[\"Cell Line\"] == cell_line]\n",
    "    # At least one that isn't Verified Normal\n",
    "    if all(data_tmp[\"Resolution_modified\"] == \"Verified Normal\") != True:\n",
    "        # First get the right data from Panel/RNA\n",
    "        if cell_line.endswith(\"T\"):\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line[:-1]]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line[:-1]]\n",
    "        else:\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line]\n",
    "\n",
    "        #Add Panel data\n",
    "        if not Panel_tmp.empty:\n",
    "            # Check if all are not VN\n",
    "            if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                # Check if multiple sequencing results for cell line\n",
    "                if len(Panel_tmp) == 1:\n",
    "                    Panel_tmp = Panel_tmp.iloc[0:, 5:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                else:\n",
    "                    # Loop through these panels\n",
    "                    Panel_tmp = create_score(Panel_tmp)\n",
    "                    Panel_tmp = Panel_tmp[Panel_tmp[\"test\"] == max(Panel_tmp[\"test\"])]\n",
    "                    if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                        if len(Panel_tmp) == 1:\n",
    "                            Panel_tmp = Panel_tmp.iloc[0:1, 5:-1]\n",
    "                            data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                        else:\n",
    "                            # Just picking the one which is not VN\n",
    "                            Panel_tmp = Panel_tmp[Panel_tmp[\"PANEL_tn_decision_clean\"] != \"VN\"]\n",
    "                            if not Panel_tmp.empty:\n",
    "                                Panel_tmp_tmp = Panel_tmp[Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VT\"]\n",
    "                                if not Panel_tmp_tmp.empty:\n",
    "                                    Panel_tmp_tmp = Panel_tmp_tmp.iloc[0:1, 5:-1]\n",
    "                                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                                else:\n",
    "                                    Panel_tmp = Panel_tmp.iloc[0:1, 5:-1]\n",
    "                                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                            else:\n",
    "                                #Add empty columns\n",
    "                                data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "                                \n",
    "                    else:\n",
    "                        #Add empty columns\n",
    "                        data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "\n",
    "            else:\n",
    "                #Add empty columns\n",
    "                data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "        \n",
    "        else:\n",
    "            #Add empty columns\n",
    "            data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "                \n",
    "        #Add RNA data\n",
    "        if not RNA_tmp.empty:\n",
    "            if len(RNA_tmp) == 1:\n",
    "                RNA_tmp = RNA_tmp.iloc[0:1, 4:]\n",
    "                data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([RNA_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "            else:\n",
    "                if len(data_tmp) == 1:\n",
    "                    RNA_tmp = RNA_tmp.iloc[0:1, 4:]\n",
    "                    data_tmp = pd.concat([pd.concat([data_tmp]*len(RNA_tmp)).reset_index(drop=True), RNA_tmp.reset_index(drop=True)], axis=1)\n",
    "                else:\n",
    "                    # Just pick one\n",
    "                    RNA_tmp = RNA_tmp.iloc[0:1, 4:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([RNA_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                        \n",
    "        else:\n",
    "            data_tmp = add_nan_RNA(data_tmp, empty_frame_RNA)\n",
    "        \n",
    "    else:\n",
    "        #Add empty columns\n",
    "        data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "        data_tmp = add_nan_RNA(data_tmp, empty_frame_RNA)\n",
    "        \n",
    "    data_tmp[\"RNA_match\"] = \"Unsure\"\n",
    "    spreadsheet.append(data_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1509,
   "metadata": {},
   "outputs": [],
   "source": [
    "spreadsheet = pd.concat(spreadsheet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1512,
   "metadata": {},
   "outputs": [],
   "source": [
    "#renaming header\n",
    "spreadsheet.columns = [\"_\".join(x.split(\" \")) for x in spreadsheet.columns]\n",
    "spreadsheet = spreadsheet.rename(columns={\"Cell_Line_Starting_Sample\":\"entity:Cell_Line_Starting_Sample_id\", \"Resolution\":\"Resolution_Original\", \"Resolution_modified\":\"Resolution\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1513,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entity:Cell_Line_Starting_Sample_id</th>\n",
       "      <th>PT-ID</th>\n",
       "      <th>SM-ID</th>\n",
       "      <th>Status</th>\n",
       "      <th>Resolution_Original</th>\n",
       "      <th>T/N</th>\n",
       "      <th>Primary_Disease</th>\n",
       "      <th>Tissue_Site</th>\n",
       "      <th>Tumor_Type</th>\n",
       "      <th>Cell_Line</th>\n",
       "      <th>...</th>\n",
       "      <th>RNA_star_bam_index</th>\n",
       "      <th>RNA_star_chimeric_bam_index</th>\n",
       "      <th>RNA_star_chimeric_junctions</th>\n",
       "      <th>RNA_star_junctions</th>\n",
       "      <th>RNA_star_junctions_pass1</th>\n",
       "      <th>RNA_star_md_bam_file</th>\n",
       "      <th>RNA_star_md_bam_index</th>\n",
       "      <th>RNA_star_read_counts</th>\n",
       "      <th>RNA_star_transcriptome_bam</th>\n",
       "      <th>RNA_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SM012T</td>\n",
       "      <td>PT-1PDR8</td>\n",
       "      <td>SM-9RFG2</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Completed</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>invasive ductal carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SM012T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DW036T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SM-CNRSK</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Completed</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DW036T</td>\n",
       "      <td>PT-21FSL</td>\n",
       "      <td></td>\n",
       "      <td>Closed</td>\n",
       "      <td>Completed</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>t-pll</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DW036T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SM-CNRSM</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Terminated</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DW036T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SM-CNRSJ</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Terminated</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JS9NN</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>CCLF_KL1338T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JT22O</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>CCLF_KL1338T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JU1FM</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>CCLF_KL1338T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JUBKV</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>CCLF_KL1338T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JUBKW</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>CCLF_KL1338T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11184 rows Ã— 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   entity:Cell_Line_Starting_Sample_id     PT-ID     SM-ID  Status  \\\n",
       "0                               SM012T  PT-1PDR8  SM-9RFG2  Closed   \n",
       "0                               DW036T       NaN  SM-CNRSK  Closed   \n",
       "1                               DW036T  PT-21FSL            Closed   \n",
       "2                               DW036T       NaN  SM-CNRSM  Closed   \n",
       "3                               DW036T       NaN  SM-CNRSJ  Closed   \n",
       "..                                 ...       ...       ...     ...   \n",
       "84                            SM-J8A8G  PT-2NPEJ  SM-JS9NN  Closed   \n",
       "85                            SM-J8A8G  PT-2NPEJ  SM-JT22O  Closed   \n",
       "86                            SM-J8A8G  PT-2NPEJ  SM-JU1FM  Closed   \n",
       "87                            SM-J8A8G  PT-2NPEJ  SM-JUBKV  Closed   \n",
       "88                            SM-J8A8G  PT-2NPEJ  SM-JUBKW  Closed   \n",
       "\n",
       "   Resolution_Original    T/N            Primary_Disease Tissue_Site  \\\n",
       "0            Completed  Tumor  invasive ductal carcinoma         NaN   \n",
       "0            Completed  Tumor                        NaN         NaN   \n",
       "1            Completed  Tumor                      t-pll         NaN   \n",
       "2           Terminated  Tumor                        NaN         NaN   \n",
       "3           Terminated  Tumor                        NaN         NaN   \n",
       "..                 ...    ...                        ...         ...   \n",
       "84      Verified Tumor  Tumor            medulloblastoma       Brain   \n",
       "85      Verified Tumor  Tumor            medulloblastoma       Brain   \n",
       "86      Verified Tumor  Tumor            medulloblastoma       Brain   \n",
       "87      Verified Tumor  Tumor            medulloblastoma       Brain   \n",
       "88      Verified Tumor  Tumor            medulloblastoma       Brain   \n",
       "\n",
       "   Tumor_Type     Cell_Line  ... RNA_star_bam_index  \\\n",
       "0         NaN       SM012T   ...                NaN   \n",
       "0         NaN        DW036T  ...                NaN   \n",
       "1         NaN        DW036T  ...                NaN   \n",
       "2         NaN        DW036T  ...                NaN   \n",
       "3         NaN        DW036T  ...                NaN   \n",
       "..        ...           ...  ...                ...   \n",
       "84    Primary  CCLF_KL1338T  ...                NaN   \n",
       "85    Primary  CCLF_KL1338T  ...                NaN   \n",
       "86    Primary  CCLF_KL1338T  ...                NaN   \n",
       "87    Primary  CCLF_KL1338T  ...                NaN   \n",
       "88    Primary  CCLF_KL1338T  ...                NaN   \n",
       "\n",
       "   RNA_star_chimeric_bam_index RNA_star_chimeric_junctions RNA_star_junctions  \\\n",
       "0                          NaN                         NaN                NaN   \n",
       "0                          NaN                         NaN                NaN   \n",
       "1                          NaN                         NaN                NaN   \n",
       "2                          NaN                         NaN                NaN   \n",
       "3                          NaN                         NaN                NaN   \n",
       "..                         ...                         ...                ...   \n",
       "84                         NaN                         NaN                NaN   \n",
       "85                         NaN                         NaN                NaN   \n",
       "86                         NaN                         NaN                NaN   \n",
       "87                         NaN                         NaN                NaN   \n",
       "88                         NaN                         NaN                NaN   \n",
       "\n",
       "   RNA_star_junctions_pass1 RNA_star_md_bam_file RNA_star_md_bam_index  \\\n",
       "0                       NaN                  NaN                   NaN   \n",
       "0                       NaN                  NaN                   NaN   \n",
       "1                       NaN                  NaN                   NaN   \n",
       "2                       NaN                  NaN                   NaN   \n",
       "3                       NaN                  NaN                   NaN   \n",
       "..                      ...                  ...                   ...   \n",
       "84                      NaN                  NaN                   NaN   \n",
       "85                      NaN                  NaN                   NaN   \n",
       "86                      NaN                  NaN                   NaN   \n",
       "87                      NaN                  NaN                   NaN   \n",
       "88                      NaN                  NaN                   NaN   \n",
       "\n",
       "    RNA_star_read_counts RNA_star_transcriptome_bam RNA_match  \n",
       "0                    NaN                        NaN    Unsure  \n",
       "0                    NaN                        NaN    Unsure  \n",
       "1                    NaN                        NaN    Unsure  \n",
       "2                    NaN                        NaN    Unsure  \n",
       "3                    NaN                        NaN    Unsure  \n",
       "..                   ...                        ...       ...  \n",
       "84                   NaN                        NaN    Unsure  \n",
       "85                   NaN                        NaN    Unsure  \n",
       "86                   NaN                        NaN    Unsure  \n",
       "87                   NaN                        NaN    Unsure  \n",
       "88                   NaN                        NaN    Unsure  \n",
       "\n",
       "[11184 rows x 57 columns]"
      ]
     },
     "execution_count": 1513,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spreadsheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1498,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create small spreadsheet if both no RNA and Panel data\n",
    "spreadsheet[\"PANEL_renamed_bai_file\"] = [str(x) for x in spreadsheet[\"PANEL_renamed_bai_file\"]]\n",
    "spreadsheet[\"RNA_fastq1\"] = [str(x) for x in spreadsheet[\"RNA_fastq1\"]]\n",
    "spreadsheet_small = spreadsheet[(spreadsheet[\"PANEL_renamed_bai_file\"] != \"nan\") | (spreadsheet[\"RNA_fastq1\"] != \"nan\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1499,
   "metadata": {},
   "outputs": [],
   "source": [
    "spreadsheet_small.to_csv(data_location / \"spreadsheet_small.tsv\", encoding = 'utf-8', sep='\\t')\n",
    "spreadsheet.to_csv(data_location / \"spreadsheet_compleet.tsv\", encoding = 'utf-8', sep='\\t')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.1 64-bit ('Broad_env': virtualenv)",
   "language": "python",
   "name": "python38164bitbroadenvvirtualenvcc06190595da421ea68a050a83be054b"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
