{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from pathlib import Path\n",
    "data_location = Path(\"../Data/Tableau_Terra_workspace\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create the terra workspace \n",
    "Read all these weird files and try to merge them to the best of my capabilities:\n",
    "\n",
    "1. Read and merge CLF with Premium\n",
    "2. Read and merge Panel data\n",
    "3. Read RNA data\n",
    "4. Read and merge WES data - UPCOMMING\n",
    "5. Apply loop through rows\n",
    "6. Duplicate rows and add conditions\n",
    "7. Check number of datapoints\n",
    "\n",
    "Note, losing alot of information due to inconsistencies in the datasets. Another note this workflow eats up memory, because I keep meta data open, just something to note\n",
    "\n",
    "### 1 Read and merge CLF with Premium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CLF data and cleaning\n",
    "CLF_data_meta = pd.read_csv(data_location / \"CLF_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "CLF_data = CLF_data_meta[[\"Cell Line\", \"PT-ID\", \"SM-ID\", \"Status\", \"Resolution\", \"T/N\", \"Primary Disease\", \"Tissue site\", \"Tumor type\", \"CELL_LINE_ID(raw)\", \"Media type\", \"Growth Properties\", \"Flask coating\", \"Starting Material Type\", \"Fail Mode\"]]\n",
    "CLF_data = CLF_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "CLF_data = CLF_data.rename(columns={\"Cell Line\": \"Cell Line Starting Sample\", \"Tissue site\": \"Tissue Site\", \"Tumor type\": \"Tumor Type\", \"CELL_LINE_ID(raw)\":\"Cell Line\", \"Growth Properties\": \"Dimension\"})\n",
    "CLF_data[\"Dataset\"] = \"CLF\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3062: DtypeWarning: Columns (5,33,80,95) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "#Premium data and cleaning\n",
    "Premium_data_meta = pd.read_csv(data_location / \"Premium_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "Premium_data = Premium_data_meta[[\"Cell Line Starting Sample\", \"PT_ID\", \"Passage Sample\", \"JIRA Status\", \"Resolution\", \"T/N\", \"Primary disease\", \"Tissue Site\", \"Tumor Type\", \"CELL_LINE_ID\", \"Growth Medium\", \"Growth Pattern\", \"ECM Bio Coatings\", \"Root Material Type\", \"Passage Sample Status\"]]\n",
    "Premium_data = Premium_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "Premium_data = Premium_data.rename(columns={\"PT_ID\": \"PT-ID\", \"Passage Sample\": \"SM-ID\", \"JIRA Status\": \"Status\", \"Primary disease\": \"Primary Disease\", \"CELL_LINE_ID\":\"Cell Line\", \"Growth Medium\":\"Media type\", \"Growth Pattern\":\"Dimension\", \"ECM Bio Coatings\":\"Flask coating\", \"Root Material Type\":\"Starting Material Type\", \"Passage Sample Status\":\"Fail Mode\"})\n",
    "Premium_data[\"Dataset\"] = \"Premium\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge CLF and Premium\n",
    "data = pd.concat([CLF_data, Premium_data])\n",
    "\n",
    "# Remove Normals\n",
    "data = data[data[\"T/N\"] == \"Tumor\"]\n",
    "\n",
    "# Cleanup some of columns\n",
    "data[\"SM-ID\"] = [re.split('\\r|  ',x)[0].strip(' \\t\\n\\r') if str(x) != 'nan' else x for x in list(data[\"SM-ID\"])]\n",
    "data[\"Resolution\"] = [str(x) for x in list(data[\"Resolution\"])]\n",
    "data[\"Cell Line\"] = [str(x) for x in list(data[\"Cell Line\"])]\n",
    "data[\"Cell Line\"] = [re.split('\\r|  ',x)[0].strip(' \\t\\n\\r') if str(x) != 'nan' else x for x in list(data[\"Cell Line\"])]\n",
    "data[\"Media type\"] = [str(x) for x in list(data[\"Media type\"])]\n",
    "data = data[data[\"Media type\"] != \"nan\"]\n",
    "\n",
    "# Note Dropping lot of conditions\n",
    "Resolution_dict = {\n",
    "    'Terminated':'Terminated',\n",
    "    'In Process':'In Process',\n",
    "    'Verification':'In Process',\n",
    "    'Ambiguous':'Drop',\n",
    "    'Completed':'Completed',\n",
    "    'Verified Normal':'Verified Normal',\n",
    "    'Verified Tumor':'Verified Tumor',\n",
    "    'Clinical team follow up':'In Process',\n",
    "    'Other verified tumor models':'Verified Tumor',\n",
    "    'Primary sample verified normal':'Verified Normal',\n",
    "    'Blacklisted; FP Failure':'Drop',\n",
    "    'Low Purity Tumor':'Drop',\n",
    "    'Clinical team follow up,Primary sample verified normal':'Verified Normal',\n",
    "    'Clinical team follow up,Other verified tumor models':'Verified Tumor',\n",
    "    'nan':'nan',\n",
    "    'Abandoned':'Drop',\n",
    "}\n",
    "\n",
    "data[\"Resolution_modified\"] = data[\"Resolution\"]\n",
    "data = data.replace({\"Resolution_modified\": Resolution_dict})\n",
    "data = data[(data[\"Resolution_modified\"] != \"Drop\") & (data[\"Resolution_modified\"] != \"In Process\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2 Read and combined Panel data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-6-dde66bea5383>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n"
     ]
    }
   ],
   "source": [
    "# TWIST Panel data\n",
    "TWIST_samples_meta = pd.read_csv(data_location / \"PANCAN_TWIST_sample.tsv\", sep=\"\\t\")\n",
    "TWIST_pair_meta = pd.read_csv(data_location / \"PANCAN_TWIST_pair.tsv\", sep=\"\\t\")\n",
    "TWIST_samples = TWIST_samples_meta[[\"entity:sample_id\", \"external_id_validation\", \"participant\", \"patient_id\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\"]]\n",
    "TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n",
    "TWIST_pair = TWIST_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "\n",
    "TWIST_samples = TWIST_samples[TWIST_samples[\"entity:sample_id\"].isin(TWIST_pair[\"case_sample\"])]\n",
    "TWIST_samples = TWIST_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TWIST = pd.merge(TWIST_samples, TWIST_pair, on='case_sample')\n",
    "TWIST = TWIST.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TWIST[\"Dataset\"] = \"TWIST\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TSCA Panel data\n",
    "TSCA_samples_meta = pd.read_csv(data_location / \"TSCA_HCMI_sample.tsv\", sep=\"\\t\")\n",
    "TSCA_pair_meta = pd.read_csv(data_location / \"TSCA_HCMI_pair.tsv\", sep=\"\\t\")\n",
    "TSCA_samples = TSCA_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_validation\", \"external_id_validation\", \"participant\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\", \"tn_decision_clean\"]]\n",
    "TSCA_pair = TSCA_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "TSCA_pair = TSCA_pair.dropna()\n",
    "\n",
    "TSCA_samples = TSCA_samples[TSCA_samples[\"entity:sample_id\"].isin(TSCA_pair[\"case_sample\"])]\n",
    "TSCA_samples = TSCA_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TSCA = pd.merge(TSCA_samples, TSCA_pair, on='case_sample')\n",
    "TSCA = TSCA.drop(columns=[\"case_sample\"])\n",
    "TSCA = TSCA.rename(columns={\"bsp_sample_id_validation\":\"case_sample\"})\n",
    "TSCA = TSCA.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TSCA[\"Dataset\"] = \"TSCA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge TWIST and TSCA with Tableau_data\n",
    "Panel = pd.concat([TWIST, TSCA])\n",
    "\n",
    "#Adding PANEL column head\n",
    "Panel = Panel.rename(columns={\"patient_id\":\"PT-ID\"})\n",
    "Panel.columns = [\"PANEL_\" + x if i > 4 else x for i,x in enumerate(Panel.columns)]\n",
    "\n",
    "Panel[\"participant\"] = [str(x) for x in Panel[\"participant\"]]\n",
    "Panel[\"PANEL_tn_decision_clean\"] = [str(x) for x in Panel[\"PANEL_tn_decision_clean\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3 Read RNA data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA data\n",
    "RNA_meta = pd.read_csv(data_location / \"CCLF_RNA_2_0.tsv\", sep=\"\\t\")\n",
    "RNA = RNA_meta[[\"entity:sample_id\", \n",
    "                     \"bsp_sample_id_rna\",  \n",
    "                     \"participant\", \n",
    "                     \"sample_type\", \n",
    "                     \"fastq1\", \n",
    "                     \"fastq2\", \n",
    "                     \"bai_file_rna\", \n",
    "                     \"bam_file_rna\", \n",
    "                     \"fusion_predictions\", \n",
    "                     \"fusion_predictions_abridged\", \n",
    "                     \"rnaseqc_count_metrics\", \n",
    "                     \"rnaseqc_count_outputs\", \n",
    "                     \"rnaseqc_exon_counts\", \n",
    "                     \"rnaseqc_gene_counts\", \n",
    "                     \"rnaseqc_gene_rpkm\", \n",
    "                     \"rsem_genes\", \n",
    "                     \"rsem_isoforms\", \n",
    "                     \"star_bam_file\", \n",
    "                     \"star_bam_index\", \n",
    "                     \"star_chimeric_bam_index\", \n",
    "                     \"star_chimeric_junctions\", \n",
    "                     \"star_junctions\", \n",
    "                     \"star_junctions_pass1\", \n",
    "                     \"star_md_bam_file\", \n",
    "                     \"star_md_bam_index\", \n",
    "                     \"star_read_counts\",\n",
    "                    \"star_transcriptome_bam\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Adding RNA column head\n",
    "RNA.columns = [\"RNA_\" + x if i > 3 else x for i,x in enumerate(RNA.columns)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-11-0eccdc508b84>:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  RNA[\"bsp_sample_id_rna\"] = [str(x) for x in RNA[\"bsp_sample_id_rna\"]]\n",
      "<ipython-input-11-0eccdc508b84>:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  RNA[\"participant\"] = [str(x) for x in RNA[\"participant\"]]\n"
     ]
    }
   ],
   "source": [
    "# Removing without SM-ID\n",
    "total_rows = len(RNA)\n",
    "RNA[\"bsp_sample_id_rna\"] = [str(x) for x in RNA[\"bsp_sample_id_rna\"]]\n",
    "RNA[\"participant\"] = [str(x) for x in RNA[\"participant\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4 Read WES data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WES data\n",
    "WES_samples_meta = pd.read_csv(data_location / \"CCLF_WES_sample.tsv\", sep=\"\\t\")\n",
    "#WES_pair_meta = pd.read_csv(data_location / \"CCLF_WES_pair.tsv\", sep=\"\\t\")\n",
    "WES_samples = WES_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_capture\", \"external_id_capture\", \"participant\", \"sample_type\", \"formatted_bai_file\", \"formatted_bam_file\", \"cnv_calls\"]]\n",
    "\n",
    "# Pair is wierd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "WES = WES_samples\n",
    "WES.columns = [\"WES_\" + x if i > 4 else x for i,x in enumerate(WES.columns)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5 Loop through data samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def growth_check(x):\n",
    "    mapping_dict = {\n",
    "        'Terminated No Growth': 0,\n",
    "        'Terminated Unlisted': 0,\n",
    "        'Terminated Fungus Contamination': 0,\n",
    "        'Terminated Stopped Growing': 0,\n",
    "        'Terminated Fungus Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,Sample Quality': 0,\n",
    "        'Terminated Fibroblast Overgrowth': 0,\n",
    "        'Terminated Feeder Contamination,No Growth': 0,\n",
    "        'Verified Normal nan': 0,\n",
    "        'Verified Normal Fibroblast Overgrowth': 0,\n",
    "        'Verified Tumor nan': 1,\n",
    "        'Verified Tumor Stopped Growing': 0,\n",
    "        'Terminated Feeder Contamination': 0,\n",
    "        'Verified Tumor Fingerprinting Failure': 2,\n",
    "        'Verified Tumor Unlisted': 0,\n",
    "        'Verified Normal Mouse Contamination': 0,\n",
    "        'Verified Normal Stopped Growing': 0,\n",
    "        'Verified Normal Unlisted': 0,\n",
    "        'Verified Normal Sample Purity': 0,\n",
    "        'Terminated Fingerprinting Failure': 2,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,No Growth': 0,\n",
    "        'Terminated No Growth,Sample Quality': 0,\n",
    "        'Terminated Sample Quality': 0,\n",
    "        'Terminated Lab Error,Sample Purity': 0,\n",
    "        'Terminated Mouse Contamination': 0,\n",
    "        'Terminated Fibroblast Overgrowth,Mouse Contamination': 0,\n",
    "        'Verified Tumor No Growth': 0,\n",
    "        'Terminated No Growth After Thaw': 0,\n",
    "        'Terminated No Growth,Sample Purity,Sample Quality': 0,\n",
    "        'Terminated Lab Error': 0,\n",
    "        'Terminated Fingerprinting Failure,Lab Error': 0,\n",
    "        'Verified Normal No Growth': 0,\n",
    "        'Terminated No Growth,Sample Purity': 0,\n",
    "        'Terminated Sample Purity': 0,\n",
    "        'Verified Tumor Terminated': 0,\n",
    "        'Terminated Terminated': 0,\n",
    "        'Terminated Received/In Container': 0,\n",
    "        'Completed Terminated': 0,\n",
    "        'Completed Received/Out of Storage': 0,\n",
    "        'Terminated nan': 0,\n",
    "        'nan nan': 2,\n",
    "        'Verified Tumor Received/Out of Storage': 2,\n",
    "        'Verified Tumor Received/In Container': 2,\n",
    "        'Verified Normal Terminated': 0,\n",
    "        'Verified Normal Received/In Container': 2,\n",
    "        'Verified Normal Received/Out of Storage': 2,\n",
    "        'nan Terminated': 0,\n",
    "        'nan Received/In Container': 2,\n",
    "        'Terminated Received/Out of Storage': 0,\n",
    "        'nan Received/Out of Storage': 2,\n",
    "        'Completed nan': 1,\n",
    "    }\n",
    "    \n",
    "    # Checking resolution and mapping back\n",
    "    x[\"Fail Mode\"] = [str(x) for x in x[\"Fail Mode\"]]\n",
    "    \n",
    "    series = x[\"Resolution_modified\"] + \" \" + x[\"Fail Mode\"]\n",
    "    series = [str(x) for x in series]\n",
    "\n",
    "    x[\"Growth\"] = series\n",
    "    x = x.replace({\"Growth\": mapping_dict})\n",
    "\n",
    "    x = x[x[\"Growth\"] != 2]\n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = growth_check(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_score(x):\n",
    "    Lookup_table = {\n",
    "        \"PRIMARY\":19,\n",
    "        \"P0\":18,\n",
    "        \"P1\":17,\n",
    "        \"P2\":16,\n",
    "        \"P3\":15,\n",
    "        \"P4\":14,\n",
    "        \"P5\":13,\n",
    "        \"P6\":12,\n",
    "        \"P7\":11,\n",
    "        \"P8\":10,\n",
    "        \"P9\":9,\n",
    "        \"P10\":8,\n",
    "        \"P11\":7,\n",
    "        \"P12\":6,\n",
    "        \"P13\":5,\n",
    "        \"P14\":4,\n",
    "        \"P15\":3,\n",
    "        \"P16\":2,\n",
    "        \"P17\":1,\n",
    "    }\n",
    "\n",
    "    test = [x.upper().split(\"_\") for x in x[\"external_id_validation\"]]\n",
    "    \n",
    "    value = []\n",
    "    for t in test:\n",
    "        for key in t:\n",
    "            if key in Lookup_table:\n",
    "                value_tmp = Lookup_table[key]\n",
    "                break\n",
    "            else:\n",
    "                value_tmp = 0\n",
    "\n",
    "        value.append(value_tmp)\n",
    "        \n",
    "    x[\"test\"] = value\n",
    "    \n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "empty_numpy_panel = np.empty([1,len(list(Panel)[5:])])\n",
    "empty_numpy_panel.fill(np.nan)\n",
    "empty_frame_panel = pd.DataFrame(empty_numpy_panel, columns=list(Panel)[5:])\n",
    "\n",
    "empty_numpy_RNA = np.empty([1,len(list(RNA)[4:])])\n",
    "empty_numpy_RNA.fill(np.nan)\n",
    "empty_frame_RNA = pd.DataFrame(empty_numpy_RNA, columns=list(RNA)[4:])\n",
    "\n",
    "empty_numpy_WES = np.empty([1,len(list(WES)[5:])])\n",
    "empty_numpy_WES.fill(np.nan)\n",
    "empty_frame_WES = pd.DataFrame(empty_numpy_WES, columns=list(WES)[5:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_nan_panel(data_tmp, empty_numpy_panel):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_panel]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)\n",
    "\n",
    "def add_nan_RNA(data_tmp, empty_frame_RNA):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_RNA]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)\n",
    "\n",
    "def add_nan_WES(data_tmp, empty_frame_WES):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_WES]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-16-c83dd44c1f94>:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  x[\"test\"] = value\n"
     ]
    }
   ],
   "source": [
    "spreadsheet = []\n",
    "\n",
    "for i, cell_line in enumerate(data[\"Cell Line\"].unique()):\n",
    "    data_tmp = data[data[\"Cell Line\"] == cell_line]\n",
    "    # At least one that isn't Verified Normal\n",
    "    if all(data_tmp[\"Resolution_modified\"] == \"Verified Normal\") != True:\n",
    "        # First get the right data from Panel/RNA\n",
    "        if cell_line.endswith(\"T\"):\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line[:-1]]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line[:-1]]\n",
    "            WES_tmp = WES[WES[\"participant\"] == cell_line[:-1]]\n",
    "        else:\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line]\n",
    "            WES_tmp = WES[WES[\"participant\"] == cell_line]\n",
    "\n",
    "        #Add Panel data\n",
    "        if not Panel_tmp.empty:\n",
    "            # Check if all are not VN\n",
    "            if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                # Check if multiple sequencing results for cell line\n",
    "                if len(Panel_tmp) == 1:\n",
    "                    Panel_tmp = Panel_tmp.iloc[0:, 5:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                else:\n",
    "                    # Loop through these panels\n",
    "                    Panel_tmp = create_score(Panel_tmp)\n",
    "                    Panel_tmp = Panel_tmp[Panel_tmp[\"test\"] == max(Panel_tmp[\"test\"])]\n",
    "                    if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                        if len(Panel_tmp) == 1:\n",
    "                            Panel_tmp = Panel_tmp.iloc[0:1, 5:-1]\n",
    "                            data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                        else:\n",
    "                            # Just picking the one which is not VN\n",
    "                            Panel_tmp = Panel_tmp[Panel_tmp[\"PANEL_tn_decision_clean\"] != \"VN\"]\n",
    "                            if not Panel_tmp.empty:\n",
    "                                Panel_tmp_tmp = Panel_tmp[Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VT\"]\n",
    "                                if not Panel_tmp_tmp.empty:\n",
    "                                    Panel_tmp_tmp = Panel_tmp_tmp.iloc[0:1, 5:-1]\n",
    "                                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                                else:\n",
    "                                    Panel_tmp = Panel_tmp.iloc[0:1, 5:-1]\n",
    "                                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                            else:\n",
    "                                #Add empty columns\n",
    "                                data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "                                \n",
    "                    else:\n",
    "                        #Add empty columns\n",
    "                        data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "\n",
    "            else:\n",
    "                #Add empty columns\n",
    "                data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "        \n",
    "        else:\n",
    "            #Add empty columns\n",
    "            data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "                \n",
    "        #Add RNA data\n",
    "        if not RNA_tmp.empty:\n",
    "            if len(RNA_tmp) == 1:\n",
    "                RNA_tmp = RNA_tmp.iloc[0:1, 4:]\n",
    "                data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([RNA_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "            else:\n",
    "                if len(data_tmp) == 1:\n",
    "                    RNA_tmp = RNA_tmp.iloc[0:1, 4:]\n",
    "                    data_tmp = pd.concat([pd.concat([data_tmp]*len(RNA_tmp)).reset_index(drop=True), RNA_tmp.reset_index(drop=True)], axis=1)\n",
    "                else:\n",
    "                    # Just pick one\n",
    "                    RNA_tmp = RNA_tmp.iloc[0:1, 4:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([RNA_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                        \n",
    "        else:\n",
    "            data_tmp = add_nan_RNA(data_tmp, empty_frame_RNA)\n",
    "                          \n",
    "        #Add WES data\n",
    "        if not WES_tmp.empty:\n",
    "            if len(WES_tmp) == 1:\n",
    "                WES_tmp = WES_tmp.iloc[0:1, 5:]\n",
    "                data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([WES_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "            else:\n",
    "                if len(data_tmp) == 1:\n",
    "                    WES_tmp = WES_tmp.iloc[0:1, 5:]\n",
    "                    data_tmp = pd.concat([pd.concat([data_tmp]*len(WES_tmp)).reset_index(drop=True), WES_tmp.reset_index(drop=True)], axis=1)\n",
    "                else:\n",
    "                    # Just pick one\n",
    "                    WES_tmp = WES_tmp.iloc[0:1, 5:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([WES_tmp]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "                        \n",
    "        else:\n",
    "            data_tmp = add_nan_WES(data_tmp, empty_frame_WES)\n",
    "        \n",
    "    else:\n",
    "        #Add empty columns\n",
    "        data_tmp = add_nan_panel(data_tmp, empty_frame_panel)\n",
    "        data_tmp = add_nan_RNA(data_tmp, empty_frame_RNA)\n",
    "        data_tmp = add_nan_WES(data_tmp, empty_frame_WES)\n",
    "        \n",
    "    data_tmp[\"RNA_match\"] = \"Unsure\"\n",
    "    spreadsheet.append(data_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "spreadsheet = pd.concat(spreadsheet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "#renaming header\n",
    "spreadsheet.columns = [\"_\".join(x.split(\" \")) for x in spreadsheet.columns]\n",
    "spreadsheet = spreadsheet.rename(columns={\"Resolution\":\"Resolution_Original\", \"Resolution_modified\":\"Resolution\", \"T/N\":\"Type\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create small spreadsheet if both no RNA and Panel data\n",
    "spreadsheet[\"PANEL_renamed_bai_file\"] = [str(x) for x in spreadsheet[\"PANEL_renamed_bai_file\"]]\n",
    "spreadsheet[\"RNA_fastq1\"] = [str(x) for x in spreadsheet[\"RNA_fastq1\"]]\n",
    "spreadsheet_small = spreadsheet[(spreadsheet[\"PANEL_renamed_bai_file\"] != \"nan\") | (spreadsheet[\"RNA_fastq1\"] != \"nan\")]\n",
    "new_index = [spreadsheet_small.iloc[i][\"Cell_Line_Starting_Sample\"] + \"_\" + str(x) for i,x in enumerate(spreadsheet_small.index)]\n",
    "spreadsheet_small.index = new_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "spreadsheet_small.to_csv(data_location / \"spreadsheet_small.tsv\", encoding = 'utf-8', sep='\\t')\n",
    "spreadsheet.to_csv(data_location / \"spreadsheet_compleet.tsv\", encoding = 'utf-8', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RNA/WES data for Gwen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1579,
   "metadata": {},
   "outputs": [],
   "source": [
    "Gwen_meta_data = data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1580,
   "metadata": {},
   "outputs": [],
   "source": [
    "Gwen_meta_data[\"participant\"] = [x[:-1] if x.endswith(\"T\") else x for x in data[\"Cell Line\"]]\n",
    "Gwen_meta_data = Gwen_meta_data[[\"Primary Disease\", \"Tissue Site\", \"Tumor Type\", \"Starting Material Type\", \"Dimension\", \"participant\"]]\n",
    "Gwen_meta_data = Gwen_meta_data.drop_duplicates(keep=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1586,
   "metadata": {},
   "outputs": [],
   "source": [
    "Gwen_RNA = pd.merge(Gwen_meta_data, RNA_meta, how='inner', on='participant')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1587,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Primary Disease</th>\n",
       "      <th>Tissue Site</th>\n",
       "      <th>Tumor Type</th>\n",
       "      <th>Starting Material Type</th>\n",
       "      <th>Dimension</th>\n",
       "      <th>participant</th>\n",
       "      <th>entity:sample_id</th>\n",
       "      <th>aggregation_product_name_rna</th>\n",
       "      <th>bai_file_rna</th>\n",
       "      <th>bam_file_rna</th>\n",
       "      <th>...</th>\n",
       "      <th>fig_random</th>\n",
       "      <th>individual_alias</th>\n",
       "      <th>rnaseqc2_exon_counts</th>\n",
       "      <th>rnaseqc2_gene_counts</th>\n",
       "      <th>rnaseqc2_gene_tpm</th>\n",
       "      <th>rnaseqc2_insertsize_distr</th>\n",
       "      <th>rnaseqc2_metrics</th>\n",
       "      <th>sample_PDO</th>\n",
       "      <th>sequencing_technology</th>\n",
       "      <th>external_rna_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SP025</td>\n",
       "      <td>SP025-Tumor-SM-C1JOO</td>\n",
       "      <td>Tru-Seq Strand Specific Large Insert RNA Seque...</td>\n",
       "      <td>gs://fc-020eb013-7449-4da7-8c7f-98294243830a/S...</td>\n",
       "      <td>gs://fc-020eb013-7449-4da7-8c7f-98294243830a/S...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SP025</td>\n",
       "      <td>SP025-Tumor-SM-HPY1Q</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SP025</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>PDO-16164</td>\n",
       "      <td>ICE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>t-pll</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Whole Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036</td>\n",
       "      <td>RP-571_DW036T_v2_RNA_OnPrem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DW036T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>melanoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CY001</td>\n",
       "      <td>CY001-Tumor-SM-B288N</td>\n",
       "      <td>Tru-Seq Strand Specific Large Insert RNA Seque...</td>\n",
       "      <td>gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...</td>\n",
       "      <td>gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>anaplastic thyroid cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pleural Effusion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JL16</td>\n",
       "      <td>CCLF_Mix_RNASeq_June2016-JL16-Tumor-SM-5LV2J</td>\n",
       "      <td>Tru-Seq Strand Specific Large Insert RNA Seque...</td>\n",
       "      <td>gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...</td>\n",
       "      <td>gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>esophageal carcinoma</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>3D</td>\n",
       "      <td>CCLF_AB1017</td>\n",
       "      <td>RP-571_CCLF_AB1017T_v1_RNA_OnPrem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_AB1017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_AB1017T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>stomach adenocarcinoma</td>\n",
       "      <td>Stomach</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>3D</td>\n",
       "      <td>CCLF_AB1030</td>\n",
       "      <td>RP-571_CCLF_AB1030T_v1_RNA_OnPrem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_AB1030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_AB1030T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>alveolar rhabdomyosarcoma</td>\n",
       "      <td>Leg</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>2D</td>\n",
       "      <td>CCLF_PEDS1064</td>\n",
       "      <td>RP-571_CCLF_PEDS1064T_v1_RNA_OnPrem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-bdfe4c68-a2bc-41b0-ac15-931e84fb1092/C...</td>\n",
       "      <td>gs://fc-bdfe4c68-a2bc-41b0-ac15-931e84fb1092/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_PEDS1064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_PEDS1064T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>squamous cell carcinoma</td>\n",
       "      <td>Head and neck region</td>\n",
       "      <td>Primary</td>\n",
       "      <td>PDX Tissue</td>\n",
       "      <td>2D</td>\n",
       "      <td>HG002</td>\n",
       "      <td>HG002-Tumor-SM-HPY1X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HG002</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...</td>\n",
       "      <td>PDO-16164</td>\n",
       "      <td>ICE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>stomach adenocarcinoma</td>\n",
       "      <td>Stomach</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Fresh Tissue</td>\n",
       "      <td>3D</td>\n",
       "      <td>CCLF_AB1050</td>\n",
       "      <td>RP-571_CCLF_AB1068T_ASC_v1_RNA_OnPrem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_AB1050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCLF_AB1068T_ASC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Primary Disease           Tissue Site Tumor Type  \\\n",
       "0                         NaN                   NaN        NaN   \n",
       "1                         NaN                   NaN        NaN   \n",
       "2                       t-pll                   NaN        NaN   \n",
       "3                    melanoma                   NaN        NaN   \n",
       "4   anaplastic thyroid cancer                   NaN        NaN   \n",
       "..                        ...                   ...        ...   \n",
       "95       esophageal carcinoma             Esophagus    Primary   \n",
       "96     stomach adenocarcinoma               Stomach    Primary   \n",
       "97  alveolar rhabdomyosarcoma                   Leg    Primary   \n",
       "98    squamous cell carcinoma  Head and neck region    Primary   \n",
       "99     stomach adenocarcinoma               Stomach    Primary   \n",
       "\n",
       "   Starting Material Type Dimension    participant  \\\n",
       "0            Fresh Tissue       NaN          SP025   \n",
       "1            Fresh Tissue       NaN          SP025   \n",
       "2             Whole Blood       NaN          DW036   \n",
       "3            Fresh Tissue       NaN          CY001   \n",
       "4        Pleural Effusion       NaN           JL16   \n",
       "..                    ...       ...            ...   \n",
       "95           Fresh Tissue        3D    CCLF_AB1017   \n",
       "96           Fresh Tissue        3D    CCLF_AB1030   \n",
       "97           Fresh Tissue        2D  CCLF_PEDS1064   \n",
       "98             PDX Tissue        2D          HG002   \n",
       "99           Fresh Tissue        3D    CCLF_AB1050   \n",
       "\n",
       "                                entity:sample_id  \\\n",
       "0                           SP025-Tumor-SM-C1JOO   \n",
       "1                           SP025-Tumor-SM-HPY1Q   \n",
       "2                    RP-571_DW036T_v2_RNA_OnPrem   \n",
       "3                           CY001-Tumor-SM-B288N   \n",
       "4   CCLF_Mix_RNASeq_June2016-JL16-Tumor-SM-5LV2J   \n",
       "..                                           ...   \n",
       "95             RP-571_CCLF_AB1017T_v1_RNA_OnPrem   \n",
       "96             RP-571_CCLF_AB1030T_v1_RNA_OnPrem   \n",
       "97           RP-571_CCLF_PEDS1064T_v1_RNA_OnPrem   \n",
       "98                          HG002-Tumor-SM-HPY1X   \n",
       "99         RP-571_CCLF_AB1068T_ASC_v1_RNA_OnPrem   \n",
       "\n",
       "                         aggregation_product_name_rna  \\\n",
       "0   Tru-Seq Strand Specific Large Insert RNA Seque...   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3   Tru-Seq Strand Specific Large Insert RNA Seque...   \n",
       "4   Tru-Seq Strand Specific Large Insert RNA Seque...   \n",
       "..                                                ...   \n",
       "95                                                NaN   \n",
       "96                                                NaN   \n",
       "97                                                NaN   \n",
       "98                                                NaN   \n",
       "99                                                NaN   \n",
       "\n",
       "                                         bai_file_rna  \\\n",
       "0   gs://fc-020eb013-7449-4da7-8c7f-98294243830a/S...   \n",
       "1   gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...   \n",
       "2   gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...   \n",
       "3   gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...   \n",
       "4   gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...   \n",
       "..                                                ...   \n",
       "95  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...   \n",
       "96  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...   \n",
       "97  gs://fc-bdfe4c68-a2bc-41b0-ac15-931e84fb1092/C...   \n",
       "98  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...   \n",
       "99  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...   \n",
       "\n",
       "                                         bam_file_rna  ... fig_random  \\\n",
       "0   gs://fc-020eb013-7449-4da7-8c7f-98294243830a/S...  ...        NaN   \n",
       "1   gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...  ...        NaN   \n",
       "2   gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...  ...        NaN   \n",
       "3   gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...  ...        NaN   \n",
       "4   gs://fc-020eb013-7449-4da7-8c7f-98294243830a/C...  ...        NaN   \n",
       "..                                                ...  ...        ...   \n",
       "95  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...  ...        NaN   \n",
       "96  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...  ...        NaN   \n",
       "97  gs://fc-bdfe4c68-a2bc-41b0-ac15-931e84fb1092/C...  ...        NaN   \n",
       "98  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...  ...        NaN   \n",
       "99  gs://fc-1d1574d1-5469-4d1d-b11d-ea13b2f0d5d9/C...  ...        NaN   \n",
       "\n",
       "   individual_alias                               rnaseqc2_exon_counts  \\\n",
       "0               NaN                                                NaN   \n",
       "1             SP025  gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "2             DW036                                                NaN   \n",
       "3               NaN                                                NaN   \n",
       "4               NaN                                                NaN   \n",
       "..              ...                                                ...   \n",
       "95      CCLF_AB1017                                                NaN   \n",
       "96      CCLF_AB1030                                                NaN   \n",
       "97    CCLF_PEDS1064                                                NaN   \n",
       "98            HG002  gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "99      CCLF_AB1050                                                NaN   \n",
       "\n",
       "                                 rnaseqc2_gene_counts  \\\n",
       "0                                                 NaN   \n",
       "1   gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "..                                                ...   \n",
       "95                                                NaN   \n",
       "96                                                NaN   \n",
       "97                                                NaN   \n",
       "98  gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "99                                                NaN   \n",
       "\n",
       "                                    rnaseqc2_gene_tpm  \\\n",
       "0                                                 NaN   \n",
       "1   gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "..                                                ...   \n",
       "95                                                NaN   \n",
       "96                                                NaN   \n",
       "97                                                NaN   \n",
       "98  gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "99                                                NaN   \n",
       "\n",
       "                            rnaseqc2_insertsize_distr  \\\n",
       "0                                                 NaN   \n",
       "1   gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "..                                                ...   \n",
       "95                                                NaN   \n",
       "96                                                NaN   \n",
       "97                                                NaN   \n",
       "98  gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...   \n",
       "99                                                NaN   \n",
       "\n",
       "                                     rnaseqc2_metrics sample_PDO  \\\n",
       "0                                                 NaN        NaN   \n",
       "1   gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...  PDO-16164   \n",
       "2                                                 NaN        NaN   \n",
       "3                                                 NaN        NaN   \n",
       "4                                                 NaN        NaN   \n",
       "..                                                ...        ...   \n",
       "95                                                NaN        NaN   \n",
       "96                                                NaN        NaN   \n",
       "97                                                NaN        NaN   \n",
       "98  gs://fc-b3a85406-97fd-452d-bbff-a44c92749ada/f...  PDO-16164   \n",
       "99                                                NaN        NaN   \n",
       "\n",
       "   sequencing_technology   external_rna_id  \n",
       "0                    NaN               NaN  \n",
       "1                    ICE               NaN  \n",
       "2                    NaN            DW036T  \n",
       "3                    NaN               NaN  \n",
       "4                    NaN               NaN  \n",
       "..                   ...               ...  \n",
       "95                   NaN      CCLF_AB1017T  \n",
       "96                   NaN      CCLF_AB1030T  \n",
       "97                   NaN    CCLF_PEDS1064T  \n",
       "98                   ICE               NaN  \n",
       "99                   NaN  CCLF_AB1068T_ASC  \n",
       "\n",
       "[100 rows x 61 columns]"
      ]
     },
     "execution_count": 1587,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Gwen_RNA"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.1 64-bit ('Broad_env': virtualenv)",
   "language": "python",
   "name": "python38164bitbroadenvvirtualenvcc06190595da421ea68a050a83be054b"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
