{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1274,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from pathlib import Path\n",
    "data_location = Path(\"../Data/Tableau_Terra_workspace\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create the terra workspace \n",
    "Read all these weird files and try to merge them to the best of my capabilities:\n",
    "\n",
    "1. Read and merge CLF with Premium\n",
    "2. Read and merge Panel data\n",
    "3. Read RNA data\n",
    "4. Read and merge WES data - UPCOMMING\n",
    "5. Apply loop through rows\n",
    "6. Duplicate rows and add conditions\n",
    "7. Check number of datapoints\n",
    "\n",
    "Note, losing alot of information due to inconsistencies in the datasets. Another note this workflow eats up memory, because I keep meta data open, just something to note\n",
    "\n",
    "### 1 Read and merge CLF with Premium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1152,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CLF data and cleaning\n",
    "CLF_data_meta = pd.read_csv(data_location / \"CLF_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "CLF_data = CLF_data_meta[[\"Cell Line\", \"PT-ID\", \"SM-ID\", \"Status\", \"Resolution\", \"T/N\", \"Primary Disease\", \"Tissue site\", \"Tumor type\", \"CELL_LINE_ID(raw)\", \"Media type\", \"Growth Properties\", \"Flask coating\", \"Starting Material Type\", \"Fail Mode\"]]\n",
    "CLF_data = CLF_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "CLF_data = CLF_data.rename(columns={\"Cell Line\": \"Cell Line Starting Sample\", \"Tissue site\": \"Tissue Site\", \"Tumor type\": \"Tumor Type\", \"CELL_LINE_ID(raw)\":\"Cell Line\", \"Growth Properties\": \"Dimension\"})\n",
    "CLF_data[\"Dataset\"] = \"CLF\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1153,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3062: DtypeWarning: Columns (5,33,80,95) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "#Premium data and cleaning\n",
    "Premium_data_meta = pd.read_csv(data_location / \"Premium_Cohort_Summary_full_data.csv\", encoding = 'utf-8', sep='\\t')\n",
    "Premium_data = Premium_data_meta[[\"Cell Line Starting Sample\", \"PT_ID\", \"Passage Sample\", \"JIRA Status\", \"Resolution\", \"T/N\", \"Primary disease\", \"Tissue Site\", \"Tumor Type\", \"CELL_LINE_ID\", \"Growth Medium\", \"Growth Pattern\", \"ECM Bio Coatings\", \"Root Material Type\", \"Passage Sample Status\"]]\n",
    "Premium_data = Premium_data.replace(r'^\\s*$', np.nan, regex=True)\n",
    "Premium_data = Premium_data.rename(columns={\"PT_ID\": \"PT-ID\", \"Passage Sample\": \"SM-ID\", \"JIRA Status\": \"Status\", \"Primary disease\": \"Primary Disease\", \"CELL_LINE_ID\":\"Cell Line\", \"Growth Medium\":\"Media type\", \"Growth Pattern\":\"Dimension\", \"ECM Bio Coatings\":\"Flask coating\", \"Root Material Type\":\"Starting Material Type\", \"Passage Sample Status\":\"Fail Mode\"})\n",
    "Premium_data[\"Dataset\"] = \"Premium\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1154,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge CLF and Premium\n",
    "data = pd.concat([CLF_data, Premium_data])\n",
    "\n",
    "# Remove Normals\n",
    "data = data[data[\"T/N\"] == \"Tumor\"]\n",
    "\n",
    "# Cleanup some of columns\n",
    "data[\"SM-ID\"] = [re.split('\\r|  ',x)[0].strip(' \\t\\n\\r') if str(x) != 'nan' else x for x in list(data[\"SM-ID\"])]\n",
    "data[\"Resolution\"] = [str(x) for x in list(data[\"Resolution\"])]\n",
    "data[\"Cell Line\"] = [str(x) for x in list(data[\"Cell Line\"])]\n",
    "data[\"Media type\"] = [str(x) for x in list(data[\"Media type\"])]\n",
    "data = data[data[\"Media type\"] != \"nan\"]\n",
    "\n",
    "# Note Dropping lot of conditions\n",
    "Resolution_dict = {\n",
    "    'Terminated':'Terminated',\n",
    "    'In Process':'In Process',\n",
    "    'Verification':'In Process',\n",
    "    'Ambiguous':'Drop',\n",
    "    'Completed':'Completed',\n",
    "    'Verified Normal':'Verified Normal',\n",
    "    'Verified Tumor':'Verified Tumor',\n",
    "    'Clinical team follow up':'In Process',\n",
    "    'Other verified tumor models':'Verified Tumor',\n",
    "    'Primary sample verified normal':'Verified Normal',\n",
    "    'Blacklisted; FP Failure':'Drop',\n",
    "    'Low Purity Tumor':'Drop',\n",
    "    'Clinical team follow up,Primary sample verified normal':'Verified Normal',\n",
    "    'Clinical team follow up,Other verified tumor models':'Verified Tumor',\n",
    "    'nan':'nan',\n",
    "    'Abandoned':'Drop',\n",
    "}\n",
    "\n",
    "data[\"Resolution_modified\"] = data[\"Resolution\"]\n",
    "data = data.replace({\"Resolution_modified\": Resolution_dict})\n",
    "data = data[(data[\"Resolution_modified\"] != \"Drop\") & (data[\"Resolution_modified\"] != \"In Process\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2 Read and combined Panel data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 925,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-925-dde66bea5383>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n"
     ]
    }
   ],
   "source": [
    "# TWIST Panel data\n",
    "TWIST_samples_meta = pd.read_csv(data_location / \"PANCAN_TWIST_sample.tsv\", sep=\"\\t\")\n",
    "TWIST_pair_meta = pd.read_csv(data_location / \"PANCAN_TWIST_pair.tsv\", sep=\"\\t\")\n",
    "TWIST_samples = TWIST_samples_meta[[\"entity:sample_id\", \"external_id_validation\", \"participant\", \"patient_id\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\"]]\n",
    "TWIST_samples['tn_decision_clean'] = \"Not Reported\"\n",
    "TWIST_pair = TWIST_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "\n",
    "TWIST_samples = TWIST_samples[TWIST_samples[\"entity:sample_id\"].isin(TWIST_pair[\"case_sample\"])]\n",
    "TWIST_samples = TWIST_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TWIST = pd.merge(TWIST_samples, TWIST_pair, on='case_sample')\n",
    "TWIST = TWIST.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TWIST[\"Dataset\"] = \"TWIST\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 926,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TSCA Panel data\n",
    "TSCA_samples_meta = pd.read_csv(data_location / \"TSCA_HCMI_sample.tsv\", sep=\"\\t\")\n",
    "TSCA_pair_meta = pd.read_csv(data_location / \"TSCA_HCMI_pair.tsv\", sep=\"\\t\")\n",
    "TSCA_samples = TSCA_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_validation\", \"external_id_validation\", \"participant\", \"sample_type\", \"renamed_bai_file\", \"renamed_bam_file\", \"depth_of_cov_qc_result\", \"cnv_calls\", \"media\", \"tn_decision_clean\"]]\n",
    "TSCA_pair = TSCA_pair_meta[[\"case_sample\", \"filtered_variants\", \"mutect1_callstats\", \"mutect1_coveragewig\", \"mutect1_powerwig\", \"mutect1_vcf\", \"mutect2_vcf\", \"oncotated_maf_mutect1\", \"oncotated_maf_mutect2\"]]\n",
    "TSCA_pair = TSCA_pair.dropna()\n",
    "\n",
    "TSCA_samples = TSCA_samples[TSCA_samples[\"entity:sample_id\"].isin(TSCA_pair[\"case_sample\"])]\n",
    "TSCA_samples = TSCA_samples.rename(columns={\"entity:sample_id\":\"case_sample\"})\n",
    "TSCA = pd.merge(TSCA_samples, TSCA_pair, on='case_sample')\n",
    "TSCA = TSCA.drop(columns=[\"case_sample\"])\n",
    "TSCA = TSCA.rename(columns={\"bsp_sample_id_validation\":\"case_sample\"})\n",
    "TSCA = TSCA.replace(r'^\\s*$', np.nan, regex=True)\n",
    "TSCA[\"Dataset\"] = \"TSCA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 936,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge TWIST and TSCA with Tableau_data\n",
    "Panel = pd.concat([TWIST, TSCA])\n",
    "\n",
    "#Adding PANEL column head\n",
    "Panel = Panel.rename(columns={\"patient_id\":\"PT-ID\"})\n",
    "Panel.columns = [\"PANEL_\" + x if i > 4 else x for i,x in enumerate(Panel.columns)]\n",
    "\n",
    "Panel[\"participant\"] = [str(x) for x in Panel[\"participant\"]]\n",
    "Panel[\"PANEL_tn_decision_clean\"] = [str(x) for x in Panel[\"PANEL_tn_decision_clean\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3 Read RNA data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 847,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA data\n",
    "RNA_meta = pd.read_csv(data_location / \"CCLF_RNA_2_0.tsv\", sep=\"\\t\")\n",
    "RNA = RNA_meta[[\"entity:sample_id\", \n",
    "                     \"bsp_sample_id_rna\",  \n",
    "                     \"participant\", \n",
    "                     \"sample_type\", \n",
    "                     \"fastq1\", \n",
    "                     \"fastq2\", \n",
    "                     \"bai_file_rna\", \n",
    "                     \"bam_file_rna\", \n",
    "                     \"fusion_predictions\", \n",
    "                     \"fusion_predictions_abridged\", \n",
    "                     \"rnaseqc_count_metrics\", \n",
    "                     \"rnaseqc_count_outputs\", \n",
    "                     \"rnaseqc_exon_counts\", \n",
    "                     \"rnaseqc_gene_counts\", \n",
    "                     \"rnaseqc_gene_rpkm\", \n",
    "                     \"rsem_genes\", \n",
    "                     \"rsem_isoforms\", \n",
    "                     \"star_bam_file\", \n",
    "                     \"star_bam_index\", \n",
    "                     \"star_chimeric_bam_index\", \n",
    "                     \"star_chimeric_junctions\", \n",
    "                     \"star_junctions\", \n",
    "                     \"star_junctions_pass1\", \n",
    "                     \"star_md_bam_file\", \n",
    "                     \"star_md_bam_index\", \n",
    "                     \"star_read_counts\",\n",
    "                    \"star_transcriptome_bam\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 848,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Adding RNA column head\n",
    "RNA.columns = [\"RNA_\" + x if i > 3 else x for i,x in enumerate(RNA.columns)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 873,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removing without SM-ID\n",
    "total_rows = len(RNA)\n",
    "RNA[\"bsp_sample_id_rna\"] = [str(x) for x in RNA[\"bsp_sample_id_rna\"]]\n",
    "RNA[\"participant\"] = [str(x) for x in RNA[\"participant\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4 Read WES data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WES data\n",
    "WES_samples_meta = pd.read_csv(data_location / \"CCLF_WES_sample.tsv\", sep=\"\\t\")\n",
    "WES_pair_meta = pd.read_csv(data_location / \"CCLF_WES_pair.tsv\", sep=\"\\t\")\n",
    "WES_samples = WES_samples_meta[[\"entity:sample_id\", \"bsp_sample_id_capture\", \"external_id_capture\", \"participant\", \"sample_type\", \"formatted_bai_file\", \"formatted_bam_file\", \"cnv_calls\"]]\n",
    "\n",
    "# Pair is wierd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5 Loop through data samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1155,
   "metadata": {},
   "outputs": [],
   "source": [
    "def growth_check(x):\n",
    "    mapping_dict = {\n",
    "        'Terminated No Growth': 0,\n",
    "        'Terminated Unlisted': 0,\n",
    "        'Terminated Fungus Contamination': 0,\n",
    "        'Terminated Stopped Growing': 0,\n",
    "        'Terminated Fungus Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,No Growth': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,Sample Quality': 0,\n",
    "        'Terminated Fibroblast Overgrowth': 0,\n",
    "        'Terminated Feeder Contamination,No Growth': 0,\n",
    "        'Verified Normal nan': 0,\n",
    "        'Verified Normal Fibroblast Overgrowth': 0,\n",
    "        'Verified Tumor nan': 1,\n",
    "        'Verified Tumor Stopped Growing': 0,\n",
    "        'Terminated Feeder Contamination': 0,\n",
    "        'Verified Tumor Fingerprinting Failure': 2,\n",
    "        'Verified Tumor Unlisted': 0,\n",
    "        'Verified Normal Mouse Contamination': 0,\n",
    "        'Verified Normal Stopped Growing': 0,\n",
    "        'Verified Normal Unlisted': 0,\n",
    "        'Verified Normal Sample Purity': 0,\n",
    "        'Terminated Fingerprinting Failure': 2,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination': 0,\n",
    "        'Terminated Bacterial Contamination,Fungus Contamination,No Growth': 0,\n",
    "        'Terminated No Growth,Sample Quality': 0,\n",
    "        'Terminated Sample Quality': 0,\n",
    "        'Terminated Lab Error,Sample Purity': 0,\n",
    "        'Terminated Mouse Contamination': 0,\n",
    "        'Terminated Fibroblast Overgrowth,Mouse Contamination': 0,\n",
    "        'Verified Tumor No Growth': 0,\n",
    "        'Terminated No Growth After Thaw': 0,\n",
    "        'Terminated No Growth,Sample Purity,Sample Quality': 0,\n",
    "        'Terminated Lab Error': 0,\n",
    "        'Terminated Fingerprinting Failure,Lab Error': 0,\n",
    "        'Verified Normal No Growth': 0,\n",
    "        'Terminated No Growth,Sample Purity': 0,\n",
    "        'Terminated Sample Purity': 0,\n",
    "        'Verified Tumor Terminated': 0,\n",
    "        'Terminated Terminated': 0,\n",
    "        'Terminated Received/In Container': 0,\n",
    "        'Completed Terminated': 0,\n",
    "        'Completed Received/Out of Storage': 0,\n",
    "        'Terminated nan': 0,\n",
    "        'nan nan': 2,\n",
    "        'Verified Tumor Received/Out of Storage': 2,\n",
    "        'Verified Tumor Received/In Container': 2,\n",
    "        'Verified Normal Terminated': 0,\n",
    "        'Verified Normal Received/In Container': 2,\n",
    "        'Verified Normal Received/Out of Storage': 2,\n",
    "        'nan Terminated': 0,\n",
    "        'nan Received/In Container': 2,\n",
    "        'Terminated Received/Out of Storage': 0,\n",
    "        'nan Received/Out of Storage': 2,\n",
    "        'Completed nan': 1,\n",
    "    }\n",
    "    \n",
    "    # Checking resolution and mapping back\n",
    "    x[\"Fail Mode\"] = [str(x) for x in x[\"Fail Mode\"]]\n",
    "    \n",
    "    series = x[\"Resolution_modified\"] + \" \" + x[\"Fail Mode\"]\n",
    "    series = [str(x) for x in series]\n",
    "\n",
    "    x[\"Growth\"] = series\n",
    "    x = x.replace({\"Growth\": mapping_dict})\n",
    "\n",
    "    x = x[x[\"Growth\"] != 2]\n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1156,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = growth_check(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1292,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_score(x):\n",
    "    Lookup_table = {\n",
    "        \"PRIMARY\":19,\n",
    "        \"P0\":18,\n",
    "        \"P1\":17,\n",
    "        \"P2\":16,\n",
    "        \"P3\":15,\n",
    "        \"P4\":14,\n",
    "        \"P5\":13,\n",
    "        \"P6\":12,\n",
    "        \"P7\":11,\n",
    "        \"P8\":10,\n",
    "        \"P9\":9,\n",
    "        \"P10\":8,\n",
    "        \"P11\":7,\n",
    "        \"P12\":6,\n",
    "        \"P13\":5,\n",
    "        \"P14\":4,\n",
    "        \"P15\":3,\n",
    "        \"P16\":2,\n",
    "        \"P17\":1,\n",
    "    }\n",
    "\n",
    "    test = [x.upper().split(\"_\") for x in x[\"external_id_validation\"]]\n",
    "    \n",
    "    value = []\n",
    "    for t in test:\n",
    "        for key in t:\n",
    "            if key in Lookup_table:\n",
    "                value_tmp = Lookup_table[key]\n",
    "                break\n",
    "            else:\n",
    "                value_tmp = 0\n",
    "\n",
    "        value.append(value_tmp)\n",
    "        \n",
    "    x[\"test\"] = value\n",
    "    \n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1423,
   "metadata": {},
   "outputs": [],
   "source": [
    "empty_numpy_panel = np.empty([1,len(list(Panel)[5:])])\n",
    "empty_numpy_panel.fill(np.nan)\n",
    "empty_frame_panel = pd.DataFrame(empty_numpy_panel, columns=list(Panel)[5:])\n",
    "\n",
    "empty_numpy_RNA = np.empty([1,len(list(RNA)[4:])])\n",
    "empty_numpy_RNA.fill(np.nan)\n",
    "empty_frame_RNA = pd.DataFrame(empty_numpy_RNA, columns=list(RNA)[4:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1426,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RNA_fastq1</th>\n",
       "      <th>RNA_fastq2</th>\n",
       "      <th>RNA_bai_file_rna</th>\n",
       "      <th>RNA_bam_file_rna</th>\n",
       "      <th>RNA_fusion_predictions</th>\n",
       "      <th>RNA_fusion_predictions_abridged</th>\n",
       "      <th>RNA_rnaseqc_count_metrics</th>\n",
       "      <th>RNA_rnaseqc_count_outputs</th>\n",
       "      <th>RNA_rnaseqc_exon_counts</th>\n",
       "      <th>RNA_rnaseqc_gene_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>RNA_star_bam_file</th>\n",
       "      <th>RNA_star_bam_index</th>\n",
       "      <th>RNA_star_chimeric_bam_index</th>\n",
       "      <th>RNA_star_chimeric_junctions</th>\n",
       "      <th>RNA_star_junctions</th>\n",
       "      <th>RNA_star_junctions_pass1</th>\n",
       "      <th>RNA_star_md_bam_file</th>\n",
       "      <th>RNA_star_md_bam_index</th>\n",
       "      <th>RNA_star_read_counts</th>\n",
       "      <th>RNA_star_transcriptome_bam</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   RNA_fastq1  RNA_fastq2  RNA_bai_file_rna  RNA_bam_file_rna  \\\n",
       "0         NaN         NaN               NaN               NaN   \n",
       "\n",
       "   RNA_fusion_predictions  RNA_fusion_predictions_abridged  \\\n",
       "0                     NaN                              NaN   \n",
       "\n",
       "   RNA_rnaseqc_count_metrics  RNA_rnaseqc_count_outputs  \\\n",
       "0                        NaN                        NaN   \n",
       "\n",
       "   RNA_rnaseqc_exon_counts  RNA_rnaseqc_gene_counts  ...  RNA_star_bam_file  \\\n",
       "0                      NaN                      NaN  ...                NaN   \n",
       "\n",
       "   RNA_star_bam_index  RNA_star_chimeric_bam_index  \\\n",
       "0                 NaN                          NaN   \n",
       "\n",
       "   RNA_star_chimeric_junctions  RNA_star_junctions  RNA_star_junctions_pass1  \\\n",
       "0                          NaN                 NaN                       NaN   \n",
       "\n",
       "   RNA_star_md_bam_file  RNA_star_md_bam_index  RNA_star_read_counts  \\\n",
       "0                   NaN                    NaN                   NaN   \n",
       "\n",
       "   RNA_star_transcriptome_bam  \n",
       "0                         NaN  \n",
       "\n",
       "[1 rows x 23 columns]"
      ]
     },
     "execution_count": 1426,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "empty_frame_RNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1424,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_nan_panel(data_tmp, empty_numpy_panel):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_panel]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)\n",
    "\n",
    "def add_nan_RNA(data_tmp, empty_frame_RNA):\n",
    "    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([empty_frame_RNA]*len(data_tmp)).reset_index(drop=True)], axis=1)\n",
    "    return(data_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1425,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "cannot concatenate object of type '<class 'numpy.ndarray'>'; only Series and DataFrame objs are valid",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1425-3673dff0ee22>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     71\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     72\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 73\u001b[0;31m             \u001b[0mdata_tmp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madd_nan_RNA\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_tmp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mempty_numpy_panel\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-1424-1f1c18ed3422>\u001b[0m in \u001b[0;36madd_nan_RNA\u001b[0;34m(data_tmp, empty_frame_RNA)\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0madd_nan_RNA\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_tmp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mempty_frame_RNA\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m     \u001b[0mdata_tmp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdata_tmp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mempty_frame_RNA\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_tmp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m     \u001b[0;32mreturn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_tmp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36mconcat\u001b[0;34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[0m\n\u001b[1;32m    269\u001b[0m     \u001b[0mValueError\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mIndexes\u001b[0m \u001b[0mhave\u001b[0m \u001b[0moverlapping\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'a'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    270\u001b[0m     \"\"\"\n\u001b[0;32m--> 271\u001b[0;31m     op = _Concatenator(\n\u001b[0m\u001b[1;32m    272\u001b[0m         \u001b[0mobjs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    273\u001b[0m         \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)\u001b[0m\n\u001b[1;32m    355\u001b[0m                     \u001b[0;34m\"only Series and DataFrame objs are valid\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtyp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    356\u001b[0m                 )\n\u001b[0;32m--> 357\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    358\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    359\u001b[0m             \u001b[0;31m# consolidate\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: cannot concatenate object of type '<class 'numpy.ndarray'>'; only Series and DataFrame objs are valid"
     ]
    }
   ],
   "source": [
    "spreadsheet = []\n",
    "\n",
    "for i, cell_line in enumerate(data[\"Cell Line\"].unique()):\n",
    "    data_tmp = data[data[\"Cell Line\"] == cell_line]\n",
    "    # At least one that isn't Verified Normal\n",
    "    if all(data_tmp[\"Resolution_modified\"] == \"Verified Normal\") != True:\n",
    "        # First get the right data from Panel/RNA\n",
    "        if cell_line.endswith(\"T\"):\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line[:-1]]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line[:-1]]\n",
    "        else:\n",
    "            Panel_tmp = Panel[Panel[\"participant\"] == cell_line]\n",
    "            RNA_tmp = RNA[RNA[\"participant\"] == cell_line]\n",
    "\n",
    "        #Add Panel data\n",
    "        if not Panel_tmp.empty:\n",
    "            # Check if all are not VN\n",
    "            if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                # Check if multiple sequencing results for cell line\n",
    "                if len(Panel_tmp) == 1:\n",
    "                    Panel_tmp = Panel_tmp.iloc[0:, 5:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index()], axis=1)\n",
    "                    spreadsheet.append(data_tmp)\n",
    "                else:\n",
    "                    # Loop through these panels\n",
    "                    Panel_tmp = create_score(Panel_tmp)\n",
    "                    Panel_tmp = Panel_tmp[Panel_tmp[\"test\"] == max(Panel_tmp[\"test\"])]\n",
    "                    if all(Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VN\") != True:\n",
    "                        if len(Panel_tmp) == 1:\n",
    "                            Panel_tmp = Panel_tmp.iloc[0:, 5:-1]\n",
    "                            data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index()], axis=1)\n",
    "                        else:\n",
    "                            # Just picking the one which is not VN\n",
    "                            Panel_tmp = Panel_tmp[Panel_tmp[\"PANEL_tn_decision_clean\"] != \"VN\"]\n",
    "                            if not Panel_tmp.empty:\n",
    "                                Panel_tmp_tmp = Panel_tmp[Panel_tmp[\"PANEL_tn_decision_clean\"] == \"VT\"]\n",
    "                                if not Panel_tmp_tmp.empty:\n",
    "                                    Panel_tmp_tmp = Panel_tmp_tmp.iloc[0:, 5:-1]\n",
    "                                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp_tmp]*len(data_tmp)).reset_index()], axis=1)\n",
    "                                else:\n",
    "                                    Panel_tmp = Panel_tmp.iloc[0:, 5:-1]\n",
    "                                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([Panel_tmp]*len(data_tmp)).reset_index()], axis=1)\n",
    "                            else:\n",
    "                                #Add empty columns\n",
    "                                data_tmp = add_nan_panel(data_tmp, empty_numpy_panel)\n",
    "                                \n",
    "                    else:\n",
    "                        #Add empty columns\n",
    "                        data_tmp = add_nan_panel(data_tmp, empty_numpy_panel)\n",
    "\n",
    "            else:\n",
    "                #Add empty columns\n",
    "                data_tmp = add_nan_panel(data_tmp, empty_numpy_panel)\n",
    "        \n",
    "        else:\n",
    "            #Add empty columns\n",
    "            data_tmp = add_nan_panel(data_tmp, empty_numpy_panel)\n",
    "                \n",
    "        #Add RNA data\n",
    "        if not RNA_tmp.empty:\n",
    "            if len(RNA_tmp) == 1:\n",
    "                RNA_tmp = RNA_tmp.iloc[0:, 4:]\n",
    "                data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([RNA_tmp]*len(data_tmp)).reset_index()], axis=1)\n",
    "            else:\n",
    "                if len(data_tmp) == 1:\n",
    "                    data_tmp = pd.concat([pd.concat([data_tmp]*len(RNA_tmp)).reset_index(), RNA_tmp.reset_index(drop=True)], axis=1)\n",
    "                else:\n",
    "                    # Just pick one\n",
    "                    RNA_tmp = RNA_tmp.iloc[0:, 4:]\n",
    "                    data_tmp = pd.concat([data_tmp.reset_index(drop=True), pd.concat([RNA_tmp]*len(data_tmp)).reset_index()], axis=1)\n",
    "                        \n",
    "        else:\n",
    "            data_tmp = add_nan_RNA(data_tmp, empty_numpy_panel)\n",
    "                \n",
    "            \n",
    "    else:\n",
    "        #Add empty columns\n",
    "        data_tmp = add_nan_panel(data_tmp, empty_numpy_panel)\n",
    "        data_tmp = add_nan_RNA(data_tmp, empty_numpy_RNA)\n",
    "            \n",
    "    data_tmp[\"RNA_match\"] = \"Unsure\"\n",
    "    spreadsheet.append(data_tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1420,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Plan shapes are not aligned",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1420-e3a52f95364b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mspreadsheet\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mspreadsheet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36mconcat\u001b[0;34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[0m\n\u001b[1;32m    282\u001b[0m     )\n\u001b[1;32m    283\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 284\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    285\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    286\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    494\u001b[0m                 \u001b[0mmgrs_indexers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    495\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 496\u001b[0;31m             new_data = concatenate_block_managers(\n\u001b[0m\u001b[1;32m    497\u001b[0m                 \u001b[0mmgrs_indexers\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnew_axes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconcat_axis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    498\u001b[0m             )\n",
      "\u001b[0;32m/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mconcatenate_block_managers\u001b[0;34m(mgrs_indexers, axes, concat_axis, copy)\u001b[0m\n\u001b[1;32m   2004\u001b[0m     \u001b[0mblocks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2005\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2006\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mplacement\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mjoin_units\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mconcat_plan\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2007\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2008\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjoin_units\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m1\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mjoin_units\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/envs/Broad_env/lib/python3.8/site-packages/pandas/core/internals/concat.py\u001b[0m in \u001b[0;36mcombine_concat_plans\u001b[0;34m(plans, concat_axis)\u001b[0m\n\u001b[1;32m    468\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mnum_ended\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnext_items\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    469\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mnum_ended\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 470\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Plan shapes are not aligned\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    471\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    472\u001b[0m             \u001b[0mplacements\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0munits\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mnext_items\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Plan shapes are not aligned"
     ]
    }
   ],
   "source": [
    "spreadsheet = pd.concat(spreadsheet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "spreadsheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1421,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_0</th>\n",
       "      <th>Cell Line Starting Sample</th>\n",
       "      <th>PT-ID</th>\n",
       "      <th>SM-ID</th>\n",
       "      <th>Status</th>\n",
       "      <th>Resolution</th>\n",
       "      <th>T/N</th>\n",
       "      <th>Primary Disease</th>\n",
       "      <th>Tissue Site</th>\n",
       "      <th>Tumor Type</th>\n",
       "      <th>...</th>\n",
       "      <th>PANEL_filtered_variants</th>\n",
       "      <th>PANEL_mutect1_callstats</th>\n",
       "      <th>PANEL_mutect1_coveragewig</th>\n",
       "      <th>PANEL_mutect1_powerwig</th>\n",
       "      <th>PANEL_mutect1_vcf</th>\n",
       "      <th>PANEL_mutect2_vcf</th>\n",
       "      <th>PANEL_oncotated_maf_mutect1</th>\n",
       "      <th>PANEL_oncotated_maf_mutect2</th>\n",
       "      <th>PANEL_Dataset</th>\n",
       "      <th>RNA_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JKWXQ</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JKWXV</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JKWY5</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JKWY7</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JKWYD</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>84</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JS9NN</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>85</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JT22O</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>86</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JU1FM</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>87</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JUBKV</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>88</td>\n",
       "      <td>SM-J8A8G</td>\n",
       "      <td>PT-2NPEJ</td>\n",
       "      <td>SM-JUBKW</td>\n",
       "      <td>Closed</td>\n",
       "      <td>Verified Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>medulloblastoma</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Primary</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unsure</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89 rows × 52 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    level_0 Cell Line Starting Sample     PT-ID     SM-ID  Status  \\\n",
       "0         0                  SM-J8A8G  PT-2NPEJ  SM-JKWXQ  Closed   \n",
       "1         1                  SM-J8A8G  PT-2NPEJ  SM-JKWXV  Closed   \n",
       "2         2                  SM-J8A8G  PT-2NPEJ  SM-JKWY5  Closed   \n",
       "3         3                  SM-J8A8G  PT-2NPEJ  SM-JKWY7  Closed   \n",
       "4         4                  SM-J8A8G  PT-2NPEJ  SM-JKWYD  Closed   \n",
       "..      ...                       ...       ...       ...     ...   \n",
       "84       84                  SM-J8A8G  PT-2NPEJ  SM-JS9NN  Closed   \n",
       "85       85                  SM-J8A8G  PT-2NPEJ  SM-JT22O  Closed   \n",
       "86       86                  SM-J8A8G  PT-2NPEJ  SM-JU1FM  Closed   \n",
       "87       87                  SM-J8A8G  PT-2NPEJ  SM-JUBKV  Closed   \n",
       "88       88                  SM-J8A8G  PT-2NPEJ  SM-JUBKW  Closed   \n",
       "\n",
       "        Resolution    T/N  Primary Disease Tissue Site Tumor Type  ...  \\\n",
       "0   Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "1   Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "2   Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "3   Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "4   Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "..             ...    ...              ...         ...        ...  ...   \n",
       "84  Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "85  Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "86  Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "87  Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "88  Verified Tumor  Tumor  medulloblastoma       Brain    Primary  ...   \n",
       "\n",
       "   PANEL_filtered_variants PANEL_mutect1_callstats PANEL_mutect1_coveragewig  \\\n",
       "0                      NaN                     NaN                       NaN   \n",
       "1                      NaN                     NaN                       NaN   \n",
       "2                      NaN                     NaN                       NaN   \n",
       "3                      NaN                     NaN                       NaN   \n",
       "4                      NaN                     NaN                       NaN   \n",
       "..                     ...                     ...                       ...   \n",
       "84                     NaN                     NaN                       NaN   \n",
       "85                     NaN                     NaN                       NaN   \n",
       "86                     NaN                     NaN                       NaN   \n",
       "87                     NaN                     NaN                       NaN   \n",
       "88                     NaN                     NaN                       NaN   \n",
       "\n",
       "   PANEL_mutect1_powerwig PANEL_mutect1_vcf PANEL_mutect2_vcf  \\\n",
       "0                     NaN               NaN               NaN   \n",
       "1                     NaN               NaN               NaN   \n",
       "2                     NaN               NaN               NaN   \n",
       "3                     NaN               NaN               NaN   \n",
       "4                     NaN               NaN               NaN   \n",
       "..                    ...               ...               ...   \n",
       "84                    NaN               NaN               NaN   \n",
       "85                    NaN               NaN               NaN   \n",
       "86                    NaN               NaN               NaN   \n",
       "87                    NaN               NaN               NaN   \n",
       "88                    NaN               NaN               NaN   \n",
       "\n",
       "   PANEL_oncotated_maf_mutect1 PANEL_oncotated_maf_mutect2  PANEL_Dataset  \\\n",
       "0                          NaN                         NaN            NaN   \n",
       "1                          NaN                         NaN            NaN   \n",
       "2                          NaN                         NaN            NaN   \n",
       "3                          NaN                         NaN            NaN   \n",
       "4                          NaN                         NaN            NaN   \n",
       "..                         ...                         ...            ...   \n",
       "84                         NaN                         NaN            NaN   \n",
       "85                         NaN                         NaN            NaN   \n",
       "86                         NaN                         NaN            NaN   \n",
       "87                         NaN                         NaN            NaN   \n",
       "88                         NaN                         NaN            NaN   \n",
       "\n",
       "    RNA_match  \n",
       "0      Unsure  \n",
       "1      Unsure  \n",
       "2      Unsure  \n",
       "3      Unsure  \n",
       "4      Unsure  \n",
       "..        ...  \n",
       "84     Unsure  \n",
       "85     Unsure  \n",
       "86     Unsure  \n",
       "87     Unsure  \n",
       "88     Unsure  \n",
       "\n",
       "[89 rows x 52 columns]"
      ]
     },
     "execution_count": 1421,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.1 64-bit ('Broad_env': virtualenv)",
   "language": "python",
   "name": "python38164bitbroadenvvirtualenvcc06190595da421ea68a050a83be054b"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
