\documentclass[11pt,]{article}
\usepackage[left=1in,top=1in,right=1in,bottom=1in]{geometry}
\newcommand*{\authorfont}{\fontfamily{phv}\selectfont}
\usepackage[]{mathpazo}


  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}




\usepackage{abstract}
\renewcommand{\abstractname}{}    % clear the title
\renewcommand{\absnamepos}{empty} % originally center

\renewenvironment{abstract}
 {{%
    \setlength{\leftmargin}{0mm}
    \setlength{\rightmargin}{\leftmargin}%
  }%
  \relax}
 {\endlist}

\makeatletter
\def\@maketitle{%
  \newpage
%  \null
%  \vskip 2em%
%  \begin{center}%
  \let \footnote \thanks
    {\fontsize{18}{20}\selectfont\raggedright  \setlength{\parindent}{0pt} \@title \par}%
}
%\fi
\makeatother




\setcounter{secnumdepth}{0}

\usepackage{longtable,booktabs}

\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}


\title{Machine Learning Draft  }



\author{\Large D J Spaanderman\vspace{0.05in} \newline\normalsize\emph{Broad Institute of MIT and Harvard}  }


\date{}

\usepackage{titlesec}

\titleformat*{\section}{\normalsize\bfseries}
\titleformat*{\subsection}{\normalsize\itshape}
\titleformat*{\subsubsection}{\normalsize\itshape}
\titleformat*{\paragraph}{\normalsize\itshape}
\titleformat*{\subparagraph}{\normalsize\itshape}


\usepackage{natbib}
\bibliographystyle{apsr}
\usepackage[strings]{underscore} % protect underscores in most circumstances



\newtheorem{hypothesis}{Hypothesis}
\usepackage{setspace}


% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{hyperref}
\usepackage{pdflscape}
\newcommand{\blandscape}{\begin{landscape}}
\newcommand{\elandscape}{\end{landscape}}

% move the hyperref stuff down here, after header-includes, to allow for - \usepackage{hyperref}

\makeatletter
\@ifpackageloaded{hyperref}{}{%
\ifxetex
  \PassOptionsToPackage{hyphens}{url}\usepackage[setpagesize=false, % page size defined by xetex
              unicode=false, % unicode breaks when used with xetex
              xetex]{hyperref}
\else
  \PassOptionsToPackage{hyphens}{url}\usepackage[draft,unicode=true]{hyperref}
\fi
}

\@ifpackageloaded{color}{
    \PassOptionsToPackage{usenames,dvipsnames}{color}
}{%
    \usepackage[usenames,dvipsnames]{color}
}
\makeatother
\hypersetup{breaklinks=true,
            bookmarks=true,
            pdfauthor={D J Spaanderman (Broad Institute of MIT and Harvard)},
             pdfkeywords = {Cancer cell line, Predicting growth media, Machine learning, AI},  
            pdftitle={Machine Learning Draft},
            colorlinks=true,
            citecolor=blue,
            urlcolor=blue,
            linkcolor=magenta,
            pdfborder={0 0 0}}
\urlstyle{same}  % don't use monospace font for urls

% Add an option for endnotes. -----


% add tightlist ----------
\providecommand{\tightlist}{%
\setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

% add some other packages ----------

% \usepackage{multicol}
% This should regulate where figures float
% See: https://tex.stackexchange.com/questions/2275/keeping-tables-figures-close-to-where-they-are-mentioned
\usepackage[section]{placeins}


\begin{document}
	
% \pagenumbering{arabic}% resets `page` counter to 1 
%
% \maketitle

{% \usefont{T1}{pnc}{m}{n}
\setlength{\parindent}{0pt}
\thispagestyle{plain}
{\fontsize{18}{20}\selectfont\raggedright 
\maketitle  % title \par  

}

{
   \vskip 13.5pt\relax \normalsize\fontsize{11}{12} 
\textbf{\authorfont D J Spaanderman} \hskip 15pt \emph{\small Broad Institute of MIT and Harvard}   

}

}








\begin{abstract}

    \hbox{\vrule height .2pt width 39.14pc}

    \vskip 8.5pt % \small 

\noindent This document provides an draft of the machine learning part of my
Masters Major Internship, I will start by introducing the subject
including some basics about machine learning, what my lab already has
achieved and how I will go fourth in achieving our goal; predicting
tumor growth media using genomics.


\vskip 8.5pt \noindent \emph{Keywords}: Cancer cell line, Predicting growth media, Machine learning, AI \par

    \hbox{\vrule height .2pt width 39.14pc}



\end{abstract}


\vskip -8.5pt

{
\hypersetup{linkcolor=black}
\setcounter{tocdepth}{2}
% \tableofcontents
}

 % removetitleabstract

\noindent  

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

The cancer cell line factory (CCLF) an initiative of the \emph{Broad
Institute} focussess on the production of cancer models
\citep{Boehm2015}. These cancer models can be assessed for various aims.
First, comprehensively characterizing the genomes of cancer models can
infer information such as cancer dependencies, which is currently
lacking in rarer types of cancers. Secondly, any lab can have access to
these new cancer models enabeling faster and more tuned research.
Thirdly, better representing cancer models to patients can improve
treatment selection. Lastly, the scale at which these models are created
will give us new insights into standardizing research protocals, which
will be mainly the focus of this draft. Additionally, several papers
have been recently published which use models derived from CCLF
\citep[\citet{Ben-David2017}, \citet{Viswanathan2017},
\citet{Joung2017}]{Hong2016}.

Due to the nature of different tumor and tissue types, effectively
creating these cancer models harnasses many technical difficulties, one
of which being the selection of growth media. Therefore, A major
bottleneck in current procedures is the required use of media panels as
large as 64 different types in order to effectively growth cancer
models. By means of trial and error we have narrowed down the candidate
growth media's for specific tumor and tissue types which we have grown
multiple times over the years. However, rarer and less frequent grown
cancer models still requires extensive media panels. In this draft I
want to explain our pipeline, several machine learning approaches and
hypothesis how we can go about using the genomic information gathered to
train machine learning models to predict the best possible growth media.

\hypertarget{pipeline}{%
\section{Pipeline}\label{pipeline}}

CCLF receives patient samples from clinicians. In our pipeline we
currently have a total of 1559 unique patient ID, ranging from the year
2001 till 2019 (based on the cohort dashboard from Tableau). In Figure
1, I have highlighted the top 10 most occuring cancer types in our
pipeline out of a total of 223 cancer types. Total frequency for each
tumor type in our database can be found in the supplementary table 1.
Note, that due to some inconsistency in naming, multiple similar tumor
types consists in the database. Additionally, I checked the current use
of media types in the top 10 most occuring tumors (Figure 2) and present
all the media types including brief explanation in supplementary table
2. Aside from the tumor type, the patient sample can have various
biological origins (i.e.~tumor site) and can originate from primary or
metastasis. Furthermore, technical differences can also exists between
samples, such as the way the sample is recieved (i.e.~fresh tissue,
cryopreserved tissue, frozen tissue etc.), types of biosp used (pleural

\newpage
\begin{landscape}

\begin{figure}
\centering
\includegraphics{figs/unnamed-chunk-1.pdf}
\caption{Number of media's tried for top 10 tumor types}
\end{figure}

\end{landscape}
\newpage

\begin{figure}
\centering
\includegraphics{figs/unnamed-chunk-2.pdf}
\caption{Number of unique patient IDs for each tumor type in the CCLF
pipeline}
\end{figure}

\noindent effusion, needle, ascites, etc.) and from which clinician.
Together these differences create noise between samples. For instance,
it might be that tumor X growth's fine with media Y, however fails due
to fact that the starting material was frozen down. It is important to
take notes of these artifacts in the data as it can influence how
further down the road, our model interpert the data, or might even learn
features which are created due to these technical differences. Noteable,
growing tumor models is a laborious and difficult task, so even when all
conditions are right, the experiment might fail. Additionally, CCLF
growths multiple types of cancer models, such as 2D (tradtional cell
lines), 3D (organoid) and neurosphere (free floating cluster of cells),
which should also be included for distinquishing tumors as these highly
influence media condition.

Before we initialize the patient sample, first the DNA is sequenced
using a large contig panel (a set of predefined regions). It is
important to note that, during the time of CCLF this panel has been
extended to include more regions of the genome, introducing either
missing data or abundant data respectively for the old and new data if
both datasets are aggegrated. Raw sequencing reads are mapped to
reference genome GRCh37 and using
\href{https://gatk.broadinstitute.org/hc/en-us}{GATK V4} including
mutect1/2, copy number variations (CNVs), small nucleotide ploymorphisms
(SNPs), insertions and deletions (Indels) are infered. Additionally,
germline events are filtered by comparing matching mutations event in
blood sample if applicable and filtering previously identified germline
events in our pipeline.

When cancer models are finalized and have been through nurcery, genome
and RNA is sequenced and compared to infer resemblance to its original
patient sample. Currently, we neither have whole genome sequencing or
RNA sequencing data for initial patient sample. Therefore, we could
explore the possibility of using cancer model data as a manner of input
data for our machine learning model as it is in theory more compleet.
Note, that RNA data is timepoint dep endent and that culturing might
have influenced the transcriptomics.

Arguable the most important descision for our media prediction model is
the chosen input data. Some general notes to take into account when
selecting input data, the higher dimensional the data, the more complex
the model requires to be to infer features in this data. Contrary, less
preprocessing such as protein pathway analysis or mutation calling has
many advantages as an end-to-end model removes the introduction of
additional technical noise, reduces analysis time and has better
standarization capabilities. In Figure 3, I have depicted a simplified
version of our current pipeline and possible input data for our model.
This could also include biological data such as tumor type, site and
metastasis or primary, however ideally this information is not included
as it is often unknown or highly variable. Additionally, multiple data
entries could be assessed as input data, however will require a more
complex model. For all input possibility I have highlighted challenges
and strengths.

\begin{figure}
\centering
\includegraphics{figs/Diagram.png}
\caption{Pipeline of CCLF and possible data type entries for our machine
learning application}
\end{figure}

\newpage

\begin{itemize}
\tightlist
\item
  \textbf{Raw sequencing reads} would create an end-to-end model without
  any preprocessing however is far to high dimensional and sparce to use
  as an input data. Implementation of raw sequencing reads have to my
  knowlegde not been done in any genomics machine learning model, aside
  from raw nanopores sequencing signal. On the other hand, mapped
  sequencing reads have been used for more elaborate tasks such as SNP
  and Indel calling, but even here this would require a far to complex
  model (i.e.~raw or mapped reads not an option as input data).
\item
  \textbf{CNV} and/or \textbf{SNPs} and/or \textbf{Indels} (and/or
  \textbf{Fusions}): with the exception of CNVs, these data entries are
  very sparce. For example, it could well be that 40 similar tumor types
  only share 5 somatic variants/Fusions, while all the other data is
  unique for each sample. An higher overview, such as mutated gene map
  or protein pathway changes could reduce this limitation. However,
  would require a number of assumptions, such as identifying important
  fusion/variant.
\item
  \textbf{RNA} have been reported in other relatable models, see the
  part literature models. Also, RNA data could better represent
  metabolic events in the data. However, RNA data is gathered from
  cancer models, which might have different transcriptomics than their
  patient sample counterpart.
\end{itemize}

\hypertarget{machine-learning}{%
\section{Machine learning}\label{machine-learning}}

Machine learning algorithms can be roughly divided between supervised
and unsupervised methods. In supervised machine learning, labeled data
is used to predict the classification or regression of data points.
Examples of `conventional' supervised machine learning algorithms are
linear and logistic regression, the random forest classifier and support
vector machines (SVM). On the other hand, in unsupervised machine
learning, patterns in data are learned without relying on predefined
labels. Two examples of unsupervised machine learning are clustering and
principal component analysis. Additionally, semi-supervised models,
which uses a combinaton of labeled and unlabeled data, in order to limit
the required learning data can also be assessed. In our case, we have
gathered a fair amount of labeled data in the form of a genomic profile
for a patient sample with both succesful and unsuccesful experiments
based on culture conditions (i.e.~Media). Therefore either a supervised
or an semi-supervised model should be assess to predict media condition.
A major bottleneck for our model is the high amount of technical noise
which consists in these experiments. \emph{``A machine learning model is
only as good as the data''} as described by
\href{http://www.research.ibm.com/5-in-5/ai-and-bias/}{IBM}.

Aside from the input data, model selection is also important. Note that
arguable the best way to approach this problem is to trial and error,
initialy starting with simpeler models such as a random forest
classifier, as most likely data is not linear seperable, ruling out
linear regression models and support vector machines. Moving forward to
more complex models such as feed-forward networks, convolutional neural
networks and recurrent neural networks. These methods are well described
in my literature study about Exploring Genomics using Deep Learning
(confident but also present in my
\href{https://github.com/Douwe-Spaanderman/Broad_DJ_AI}{GitHub
repository}, please be mindfull when sharing). This review also
describes shortly the bascis of deep learning such as back-propogation,
activation functions, loss functions, hyperparamaterization, which I
won't go into further here as it is an extensive subject. However, on
the technical aspect there are several tools we could assess to create
the best model for our needs.
\href{https://scikit-learn.org/stable/}{scikit-learn} which is an
extensive python library can be used for various machine learning
algorithms such as random forest, dimensional reduction techniques and
preprocessing (feature extraction and normalization).
\href{https://www.tensorflow.org/}{TensorFlow} and
\href{https://keras.io/}{Keras} respectively low and higher level neural
network API's to create deep learning models. Additionally,
\href{https://github.com/maxpumperla/hyperas}{Hyperas} and
\href{https://github.com/autonomio/talos}{Talos} are hyperparameter
optimization algorithms and
\href{https://github.com/dvgodoy/deepreplay}{DeepReplay} assist learning
visualization.

Aside from these supervised learning models, we could also invest time
in dimensional reduction techniques to infer relations in mutation data
as a method of pre-processing such as principal component analysis
(PCA), autoencoder or generative adversarial network (GAN).

\hypertarget{literature-models}{%
\section{Literature models}\label{literature-models}}

In my literature study, I have described many deep learning approaches
in genomics. However, few models are described to achieve similar goals
as predicting media conditions from genomic data. Nevertheless,
\citet{Lyu2018}, et al.~describes the use of gene expression data in
order to classify tumor types. In order to achieve this goal, RNA data
is log2 transformed and compared to Pan-Cancer Atlas in order to reduce
the gene panel. Next these gene panels are embedded into a 2D image
(102x102), which encodes for gene expression of 10404 important genes.
These images are than analyzed using a convolutional neural network.
Some things to note with this method, embedding is arguable the most
dominant factor in model effectiveness. This is due to the nature of
CNNs, as they take spatial dependencies in data into account. Therefore
shifting around gene position would greatly impact model accurarcy. The
main message we can take from this paper is the method they used to
encode their genomic data by selecting important genes from Pan-Cancer
Atlas and encoding into a 2D heatmap.

A more recent paper, uses a deep learning approach to classify primary
and metastatic cancer using passenger mutation patterns
\citep{Jiao2020}. somatic mutation were preprocessed to extract several
features. For each sample, the mutational-type feature was based on
counting the number of single nucleotide changes, nucleotide changes
plus their 5' and/or 3' flanking nucleotides. Next, these
mutational-type feautres were normalized for the total number of SNVs in
the sample. The mutational distribution features are the number of SNVs,
small indels, structural variation (SV) breakpoints and CNV in
1-megabase bins across the genome, normalized to the corresponding
mutational event across the genome. Additional features are the total
number of each type of mutational event per genome, number of each type
of mutational event per chromosome (normalized for chromosome size),
sample purity and sample ploidy. These features were than used in an
feed-forward network. Noteable, adding information on driver mutations
reduced model accuracy. This paper presents another method of encoding
genomic data for our machine learning model.

\hypertarget{conclusion}{%
\section{Conclusion}\label{conclusion}}

In this paragraph I will discuss the best steps to follow in order to
create a model which will be able to infer media condition from genomic
data. Currently, the main bottleneck is selecting the type of input data
and how to encode this information. Both models reported in the previous
segment show promosing methods of encoding genomic data for deep
learning approaches. In my opinion, RNA data seems more intuitive as it
closer resemblence metabolic features, which are most important for
media selection. However I am unsure if the transcriptomics data we have
is applicable for predicting media condition as it originates from
cancer models instead of patient sample. On the other hand, we have more
abundant mutation data from patient sample. Preprocessing would be a
more elaborate task when assessing mutation data. Similar to the paper
described in the previous segment, we could encode mutation data in
features. In contrast, we could encode mutation data as higher level
structures such as gene level (i.e.~gene x has y mutation data) or even
in an protein pathway (i.e.~pathway x has gene y with z mutation data).
This would reduce the effect of sparseness in mutation data, however
would require a fair amount of assumptions, such as which pathway/gene
to feed the network, if and how to include functional mutation
annotation. In order to create a working machine learning model several
steps have to be conducted:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Aggegrating compleet datasets, ideally a Terra workspace in which data
  is cleaned and displayed as Patient ID X - Biological/technical
  information (i.e.~primary-biopsy-type etc.) - Raw/mapped reads - CNV -
  SNV/indels - Media types (csv file with all media types + succes rates
  for each type) - linked cancer model id - WGS (incl CNV/SNV) - RNA
\item
  Encoding either mutation information or transcriptomics based on
  feedback from Remi/Moony
\item
  Segregation of training, test and validation dataset
\item
  Building, training and validating machine learning algorithm
\item
  Blackbox features extraction. This is something very promising if the
  algorithm works. It would possibly give insight into why a certian
  mutational/transcriptomics landscape prefers a set media type.
\end{enumerate}

Something to note is that ideally I would like to access all the created
cancer models, by removing any predefined information such as tumor type
and site. However, it also might be worthwhile to reduce noise by
removing tumors types which are only few in our pipeline (\textless{}5).

\newpage

\hypertarget{supplementary}{%
\section{Supplementary}\label{supplementary}}

\begin{longtable}[]{@{}rl@{}}
\toprule
Number of times reported & Tumor type\tabularnewline
\midrule
\endhead
1 & acinar cell carcinoma\tabularnewline
3 & adamantinomatous craniopharyngioma\tabularnewline
6 & adenocarcinoma\tabularnewline
8 & adenocarcinoma nos\tabularnewline
1 & adenoid cystic carcinoma\tabularnewline
1 & adrenal cortical neoplasm\tabularnewline
1 & adrenocortica adenoma\tabularnewline
1 & alcl\tabularnewline
9 & alveolar rhabdomyosarcoma\tabularnewline
1 & alveolar soft part sarcoma\tabularnewline
1 & aml\tabularnewline
11 & anaplastic astrocytoma\tabularnewline
4 & anaplastic ependymoma\tabularnewline
1 & anaplastic meningioma\tabularnewline
1 & anaplastic oligoastrocytoma\tabularnewline
2 & anaplastic oligodendroglioma\tabularnewline
2 & anaplastic oliogodendroglioma\tabularnewline
1 & anaplastic thryoid cancer\tabularnewline
3 & anaplastic thyroid cancer\tabularnewline
1 & anaplastic thyroid carcinoma\tabularnewline
1 & anaplastic wilms tumor\tabularnewline
1 & aneurysmal bone cyst\tabularnewline
1 & angiocentric astrocytoma\tabularnewline
1 & angiocentric glioma\tabularnewline
5 & angioimmunoblastic t-cell lymphoma\tabularnewline
1 & angiomatoid fibrous histiocytoma\tabularnewline
5 & angiosarcoma\tabularnewline
4 & astrocytoma\tabularnewline
3 & b cell lymphoma\tabularnewline
1 & bladder carcinoma\tabularnewline
2 & breast adenocarcinoma\tabularnewline
5 & breast cancer\tabularnewline
13 & breast carcinoma\tabularnewline
1 & breast poorly-differentiated carcinoma\tabularnewline
14 & carcinoma\tabularnewline
1 & cardiac undifferentiated sarcoma\tabularnewline
1 & cholangiocarcinoma\tabularnewline
1 & chondrosarcoma\tabularnewline
5 & chordoma\tabularnewline
1 & choriocarcinoma\tabularnewline
1 & chromophobe\tabularnewline
8 & chromophobe renal cell carcinoma\tabularnewline
1 & chromophobes, renal cancer\tabularnewline
1 & clear cell meningioma\tabularnewline
2 & clear cell renal cell carcinoma\tabularnewline
9 & colon adenocarcinoma\tabularnewline
1 & colon cancer\tabularnewline
3 & colorectal adenocarcinoma\tabularnewline
16 & colorectal cancer\tabularnewline
1 & congenital mesoblastic nephroma\tabularnewline
1 & congenital/infantile rhabdomyosarcoma\tabularnewline
3 & craniopharyngioma\tabularnewline
1 & cystic renal disease\tabularnewline
1 & cystic teratoma\tabularnewline
4 & desmoid tumor\tabularnewline
1 & desmoplastic small round cell blue tumor\tabularnewline
1 & desmoplastic small round cell tumor\tabularnewline
1 & differentiated thyroid cancer\tabularnewline
3 & diffuse astrocytoma\tabularnewline
1 & diffuse large b cell lymphoma\tabularnewline
3 & double-hit diffuse large b cell lymphoma\tabularnewline
1 & dysembroplastic neuroepithelial tumor\tabularnewline
1 & eatl\tabularnewline
7 & embryonal rhabdomyosarcoma\tabularnewline
1 & embryonal sarcoma\tabularnewline
1 & embryonal tumor\tabularnewline
5 & ependymoma\tabularnewline
1 & epitelioid sarcoma\tabularnewline
1 & epithelioid hemmanggioendothelioma\tabularnewline
1 & epstein-barr virus-associated smooth muscle tumor\tabularnewline
68 & esophageal adenocarcinoma\tabularnewline
31 & esophageal carcinoma\tabularnewline
4 & esophageal squamous cell carcinoma\tabularnewline
4 & esophogeal carcinoma {[}esca{]}\tabularnewline
19 & ewing sarcoma\tabularnewline
1 & extrahepatic cholangiocarcinoma\tabularnewline
2 & fibrolamellar hepatocellular carcinoma\tabularnewline
3 & ganglioglioma\tabularnewline
1 & ganglioma\tabularnewline
5 & ganglioneuroblastoma\tabularnewline
3 & ganglioneuroma\tabularnewline
52 & gastric adenocarcinoma\tabularnewline
1 & gastric cancer\tabularnewline
6 & gastric poorly-differentiated carcinoma\tabularnewline
3 & gastro-esophageal junction adenocarcinoma\tabularnewline
2 & gastroesophageal junction adenocarcinoma\tabularnewline
5 & gastrointestinal stromal tumor\tabularnewline
3 & gej adenocarcinoma\tabularnewline
3 & germ cell tumor\tabularnewline
1 & giant cell tumor\tabularnewline
26 & glioblastoma\tabularnewline
153 & glioblastoma multiforme\tabularnewline
13 & glioma\tabularnewline
17 & glioma high grade\tabularnewline
6 & glioma low grade\tabularnewline
7 & glioma nos\tabularnewline
6 & glioma, high grade\tabularnewline
1 & hemangioblastoma\tabularnewline
11 & hepatoblastoma\tabularnewline
7 & hepatocellular carcinoma\tabularnewline
1 & hgg vs gbm\tabularnewline
1 & high grade glioma\tabularnewline
1 & histiocytoma\tabularnewline
1 & hnscc\tabularnewline
1 & hstcl\tabularnewline
1 & hyalinized fibrous breast tissue consistent with treatment
effect.\tabularnewline
1 & infantile fibrosarcoma\tabularnewline
2 & intrahepatic cholangiocarcinoma\tabularnewline
1 & invasive adenocarcinoma\tabularnewline
6 & invasive carcinoma\tabularnewline
5 & invasive ductal carcinoma\tabularnewline
1 & juvenile granulosa cell tumor\tabularnewline
3 & kidney renal clear cell carcinoma\tabularnewline
1 & kidney renal clear cell carcinoma {[}kirc{]}\tabularnewline
1 & kidney renal translocation\tabularnewline
6 & lam\tabularnewline
1 & langerhans cell histiocytosis\tabularnewline
1 & large b cell non-hodgkin lymphoma\tabularnewline
6 & leiomyosarcoma\tabularnewline
4 & liposarcoma\tabularnewline
12 & lung adenocarcinoma\tabularnewline
7 & lung cancer\tabularnewline
3 & lung carcinoma\tabularnewline
1 & lung squamous cell carcinoma\tabularnewline
1 & lymphoma\tabularnewline
1 & malignant glomus\tabularnewline
9 & malignant peripheral nerve sheath tumor\tabularnewline
1 & mature cystic teratoma\tabularnewline
2 & medullary thyroid cancer\tabularnewline
1 & medullary thyroid carcinoma\tabularnewline
1 & medullary tumor\tabularnewline
6 & medulloblastoma\tabularnewline
72 & melanoma\tabularnewline
2 & melanotic neuroectodermal tumor\tabularnewline
14 & meningioma\tabularnewline
2 & metastatic adenocarcinoma\tabularnewline
7 & metastatic carcinoma\tabularnewline
1 & metastatic poorly differentiated carcinoma\tabularnewline
1 & mixed adeno-neuroendocrine (manec) of the ge junction\tabularnewline
1 & mod diff id/l ca\tabularnewline
1 & mpnst\tabularnewline
1 & mucinous cystadenoma, mature cystic teratoma\tabularnewline
1 & mullerian adenocarcinoma\tabularnewline
1 & myoepithelial cancer\tabularnewline
1 & myofibroblastic sarcoma\tabularnewline
3 & myofibroblastic tumor\tabularnewline
2 & myxoid liposarcoma\tabularnewline
2 & myxopapillary ependymoma\tabularnewline
47 & neuroblastoma\tabularnewline
2 & neuroepithelial tumor\tabularnewline
1 & neurofibroma\tabularnewline
1 & nk/t\tabularnewline
1 & non-malignant\tabularnewline
1 & non-small cell lung cancer\tabularnewline
3 & not cancer\tabularnewline
1 & nut carcinoma\tabularnewline
1 & nut midline carcinoma\tabularnewline
3 & ocular melanoma\tabularnewline
1 & olfactory neuroblastoma\tabularnewline
10 & oligodendroglioma\tabularnewline
1 & oligometastatic non-small-cell lung cancer\tabularnewline
32 & osteosarcoma\tabularnewline
2 & ovarian cancer\tabularnewline
2 & ovarian carcinoma\tabularnewline
180 & pancreatic adenocarcinoma\tabularnewline
24 & pancreatic adenocarcinoma {[}paad{]}\tabularnewline
8 & pancreatic ductal adenocarcinoma\tabularnewline
2 & pancreatic poorly-differentiated carcinoma\tabularnewline
2 & pancreatic pseudopapillary tumor\tabularnewline
5 & papillary thyroid cancer\tabularnewline
4 & papillary thyroid carcinoma\tabularnewline
1 & parotid carcinoma\tabularnewline
4 & pheochromocytoma and paraganglioma\tabularnewline
2 & pilocytic astrocytoma\tabularnewline
1 & pilomyxoid astrocytoma\tabularnewline
1 & pineal anlage tumor\tabularnewline
1 & pineoblastoma\tabularnewline
2 & pituitary adenoma\tabularnewline
12 & pituitary tumor\tabularnewline
2 & pleomorphic adenoma\tabularnewline
3 & pleuropulmonary blastoma\tabularnewline
1 & plexiform neurofibroma\tabularnewline
2 & poorly differentiated carcinoma\tabularnewline
1 & poorly-differentiated carcinoma\tabularnewline
9 & prostate adenocarcinoma\tabularnewline
39 & prostate cancer\tabularnewline
5 & ptcl-nos\tabularnewline
1 & renal carcinoma\tabularnewline
26 & renal cell carcinoma\tabularnewline
6 & renal clear cell carcinoma\tabularnewline
2 & renal medullary carcinoma\tabularnewline
3 & retinoblastoma\tabularnewline
4 & rhabdoid tumor\tabularnewline
3 & rhabdomyosarcoma\tabularnewline
2 & sarcoma\tabularnewline
2 & schawnnoma\tabularnewline
2 & schwannoma\tabularnewline
1 & sertoli-leydig tumor\tabularnewline
6 & sezary syndrome\tabularnewline
1 & sialoblastoma\tabularnewline
1 & skin cutaneous melanoma\tabularnewline
21 & skin cutaneous melanoma {[}skcm{]}\tabularnewline
1 & solid pseudopapillary tumor of the pancreas\tabularnewline
1 & solitary fibrous tumor\tabularnewline
1 & spindle cell carcinoma\tabularnewline
1 & spindle cell lesion\tabularnewline
3 & spindle cell sarcoma\tabularnewline
18 & squamous cell carcinoma\tabularnewline
48 & stomach adenocarcinoma\tabularnewline
11 & stomach adenocarcinoma {[}stad{]}\tabularnewline
1 & stomach/esophageal\tabularnewline
6 & synovial sarcoma\tabularnewline
13 & t-pll\tabularnewline
1 & testicular cancer\tabularnewline
1 & testicular germ cell\tabularnewline
1 & thymoma\tabularnewline
6 & thyroid cancer\tabularnewline
1 & thyroid carcinoma\tabularnewline
2 & undifferentiated sarcoma\tabularnewline
6 & unknown\tabularnewline
6 & urothelial carcinoma\tabularnewline
45 & wilms tumor\tabularnewline
3 & yolk sac tumor\tabularnewline
\bottomrule
\end{longtable}

Table 1 : Frequency table of the occurence of tumor types in our
pipeline, take note that some naming issues are currently present in our
pipeline, such as stomach adenocarcinoma and stomach adenocarcinoma
{[}stad{]}, which should be considered the same, therefore cleaning of
the data should be conducted to aggegrate these results together, either
by manually (cleaner) or artificially (faster) mapping these tumor
types.





\newpage
\singlespacing 
\bibliography{data/master.bib}

\end{document}
